\documentclass[12pt,notitlepage]{article}

% Overleaf project:
% https://www.overleaf.com/project/5fdb880ef955c3369964d601

% View-only link:
% https://www.overleaf.com/read/hkybwqtbtvjb

\usepackage{amsmath, amsfonts, amssymb}
\usepackage[svgnames]{xcolor}
\usepackage{datetime2}
\usepackage[
	colorlinks=true, 
	citecolor={DarkRed}, urlcolor={DarkBlue}, linkcolor={DarkBlue},
]{hyperref}


% 
\usepackage[version=4]{mhchem}

\usepackage{fullpage}

% Paragraph spacing
\usepackage{parskip}

\usepackage{xspace}

\usepackage{graphicx}
\graphicspath{{../images/}}

%\usepackage{epstopdf}

% Tables (the order matters here)
\usepackage{makecell}
\usepackage{booktabs}
\usepackage{arydshln}

% For editing purposes:
%\usepackage[margin=10pt]{geometry}

% https://latex.org/forum/viewtopic.php?t=10456
\usepackage{titlesec}
\titleformat{\subsubsection}[runin]% runin puts it in the same paragraph
{\normalfont\bfseries}% formatting commands to apply to the whole heading
{\thesubsubsection}% the label and number
{}% space between label/number and subsection title
{}% formatting commands applied just to subsection title
[.]% punctuation or other commands following subsection title


\newcommand{\TODO}[1]{\textrm{\color{red}TODO: #1}}
\input{todomech}


\renewcommand{\d}{\mathrm{d}}

\newcommand{\NOT}{\ensuremath{\mathop{\mathsf{not}}}\xspace}
\newcommand{\AND}{\ensuremath{\mathop{\mathsf{and}}}\xspace}
\newcommand{\OR}{\ensuremath{\mathop{\mathsf{or}}}\xspace}
\newcommand{\XOR}{\ensuremath{\mathop{\mathsf{xor}}}\xspace}

\newcommand{\TEXT}[1]{\quad\text{#1}\quad}
\newcommand{\with}{\text{$\,{:}\,$}}

\newcommand{\cbra}[1]{{\ensuremath{\color{gray}{#1}}}}
\newcommand{\signal}[1]{{{\cbra{\langle}\ce{#1}\cbra{\rangle}}}}
\newcommand{\protein}[1]{{{\cbra{(}\ce{#1}\cbra{)}}}}
\newcommand{\promoter}[1]{{{\cbra{[}\ce{#1}\cbra{]}}}}

% https://tex.stackexchange.com/questions/543953/arrow-with-blunted-end-head-in-math-mode
\newcommand{\act}{\ {\ensuremath{\mathbin{\to}}}\ }
\newcommand{\rep}{\ {\ensuremath{\mathrel{\raisebox{-.3ex}{\rotatebox{90}{\scalebox{1}[1.2]{$\bot$}}}}}}\ }

\def\[#1\]{\begin{align}#1\end{align}}

% https://tex.stackexchange.com/questions/114113/how-to-label-text-with-equation-number
\newcommand{\eqnum}{\leavevmode\hfill\refstepcounter{equation}\textup{{(\theequation)}}}


\newcommand{\hh}[1]{{\color{Purple}#1}}
\newcommand{\ra}[1]{{\color{Blue}#1}}


\title{ibiocomp project}
\author{RA \& HH}
\date{\today}


\begin{document}

\maketitle

\ra{LOCKED 'COS WORKING WITH ANOTHER COPY (WRITE EMAIL IF NEEDED)}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Introduction}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\subsection{The game of 15 sticks}

We design genetic circuits
that play the following game:
%
\begin{quote}
	Two players start with 15 sticks
	and alternatingly 
	take 1, 2 or 3 sticks.
	\\
	The last player to take a stick loses.
\end{quote}

Because a player can complement
the number of sticks taken by the opponent,
there is a deterministic winning strategy
of the first-mover
as shown in Table \ref{t:logical-playera}.
%
%
We implement this strategy for both players.


\subsection{General design}

We encode in binary 
the number of sticks left as
\[
    \label{e:s0123}
    %
    \text{number of sticks left} = \ce{8 s_3 + 4 s_2 + 2 s_1 + s_0}
    .
\]
%
%
By the 4-periodicity of the winning strategy,
only the two lowest bits \ce{s_1}/\ce{s_0}
matter to the player.
%
%
The response of a player is a number 1, 2, or 3,
which we encode in binary as
\[
    \label{e:r01}
    %
    \text{number of sticks taken by a player} = \ce{2 r_1 + r_0}
    .
\]
%
The bits in \eqref{e:s0123}--\eqref{e:r01}
are transmitted by 
intercellular signaling molecules,
cf.~\S\ref{ss:experiment}.
%
Although our players are essentially identical,
we model them separately and with one difference:
Player A (B) becomes active when 
the extracellular signal \ce{w_A} (\ce{w_B}) is supplied.

%

The \emph{subtractor} module
keeps track of the number of sticks left \eqref{e:s0123}.
To appreciate the key design issue, suppose
we manually transmit the response \ce{r_1 r_0 = 01}
of a player
into the medium of the subtractor.
It should compute the new number of the sticks,
yet, it has to do so once
and not keep subtracting the value \ce{r_1 r_0}
that is still present.
One solution is single-use subtractors
but then the result of each computation
has to be imparted to an unused subtractor.
Barring excessive manual intervention,
a feasible design
seems to require some separation of
reading/storing the current state,
computing the new state and communicating it
--
temporally or spatially.
For our purposes
this may include
modest parallelism in intercellular signaling
(but not frequency encoding).
%
%
We propose 
a subtractor composed of two parts A and B,
and
a workflow 
%of Phase A, Interphase, Phase B, Interphase, etc.,
as shown in Table \ref{t:workflow}.
%
%
%
Thus,
in Phase A characterized by the supply of 
the signal \ce{w_A},
Subtractor A
emits the current state \ce{s},
Player A responds with \ce{r},
while
Subtractor B
reads \ce{s}/\ce{r}
and
computes the new state % \ce{s} 
{silently}
and remembers the result
in order to emit it in the subsequent Phase B.
%
The medium is cleared, before
Phase B is initiated by
supplying the signal \ce{w_B}.
%
%
In short,
we require
a gated memory module and a delayed response,
cf.~\S\ref{s:sub}.

%

Refinements could include
a)
dealing with
a player who might attempt
to take more sticks than \eqref{e:s0123};
b)
dedicated sensor cells 
that report 
on the presence of signaling molecules 
in order to 
monitor the state of the game.

%

%%% TABLE %%%
\input{t_workflow}


\subsection{Experimental design}
\label{ss:experiment}

We distinguish
\emph{extracellular} chemical signals 
that are supplied by the experimenter
and
\emph{intercellular} signals
that are produced and sensed by cells.
%
\TODO{what are \ce{w_A} and \ce{w_B}?}
%
%
For an experiment in a single test tube 
we need
orthogonal intercellular channels:
\emph{four} for \eqref{e:s0123},
\emph{two} for \eqref{e:r01},
and
\emph{three} for the carry flags
(see \S\ref{s:sub}).
%
One channel can be spared 
at the expense
of a more involved computation of 
the last useful carry flag $c_3$.
%
Thus \emph{eight} orthogonal intercellular channels
would be required.
%
This is possible according to \cite{DuETAL2020},
where 10 intercellular signaling
pathways were designed de novo
and
four known QS pathways optimized.
%
Among those, only eight worked well orthogonally
on the sensor level 
and 
seven on the promoter level,
see
\cite[\href{https://www.nature.com/articles/s41467-020-17993-w/figures/3}{Fig.~3c/g}]{DuETAL2020}.
%
It is pointed out
\cite[p.6]{DuETAL2020}
that 
the signaling pathways 
\TODO{RpaR-Prpa* and LuxR-Plux*}
should not coexist in one cell,
which we avoid below.
%

In the interest of experimental feasibility
we partition the workflow
with spatial compartments
that require manual transmission of 
the medium.
%
%
This allows to save on another two intercellular channels.
%
%
%

\ra{Note, from \cite[SM, VII.M]{NielsenETAL2016}:
\begin{quote}
    The sensors used include a truncated AraC 
    (AraC-C280*, referred to as AraC* in this work)
    that has reduced cross-talk with IPTG(40) and its output promoter PBAD
    which is induced using L-arabinose; 
    LacI and its output promoter with a symmetric lac-operator PTac which
    is IPTG-inducible(41); 
    and TetR with its output promoter PTet(42)
\end{quote}
}

\TODO{Please check:}

\begin{itemize}
\item[\ce{w_A}:]
    $
        \href{https://pubchem.ncbi.nlm.nih.gov/compound/L-Arabinose}{\signal{Ara}}
        \act
        \protein{AraC^*}
        \act
        \promoter{BAD}
    $

\item[\ce{w_B}:]
    $
        \href{https://pubchem.ncbi.nlm.nih.gov/compound/656894}{\signal{IPTG}}
        \rep
        \protein{LacI}
        \rep
        \promoter{Tac}
    $
    
    \ra{
    is that the usual promoter controlled by LacI?
    }
    
\item[\ce{r_0}:]
    $
        \href{https://pubchem.ncbi.nlm.nih.gov/compound/119133}{\signal{{3OC6}{-}{HSL}}}
        \act
        \protein{LuxR}
        \act
        \promoter{Lux}
    $
    
    \ra{[this is already discussed below]}

\item[\ce{r_1}:]
    $
        \href{https://pubmed.ncbi.nlm.nih.gov/21949379/}{\signal{IV{-}HSL}}
        \TODO{?}
        \protein{BjaR}
        \TODO{?}
        \promoter{Bja}
    $
    
    \TODO{reference?}
    \TODO{good link to IV-HSL?}

\item[\ce{s_0}:]
    $
        \href{https://en.wikipedia.org/wiki/2,4-Diacetylphloroglucinol}{\signal{DAPG}}
        \rep
        \protein{PhlF}
        \rep
        \promoter{PhlF}
    $
    % This appears in \cite{NielsenETAL2016}
 
    Note: \protein{PhlF} used a lot in \cite{NielsenETAL2016}
    
\item[\ce{c_1}:]
    $
        \signal{Sal}
        \act
        \protein{NahR}
        \act
        \promoter{Sal}
    $
    
    \ra{[this is already discussed below]}
    
    \TODO{what exactly is \href{https://en.wikipedia.org/wiki/Salicylic_acid}{Sal}?}
    
\item[\ce{s_1}:]
    $
        \href{https://pubchem.ncbi.nlm.nih.gov/compound/N-_4-Coumaroyl_-L-homoserine-lactone}{\signal{pC{-}HSL}}
        \act
        \protein{RpaR}
        \act
        \promoter{Rpa}
    $

    \ra{[this is already discussed below]}
    
\item[\ce{c_2}:]
    $
        \signal{MMF}
        \TODO{?}
        \protein{MmfR}
        \TODO{?}
        \promoter{Mmf}
    $
    
    \TODO{
    what is MMF?
    cf.~\url{https://pubmed.ncbi.nlm.nih.gov/19054329/}
    }

\item[\ce{s_2}:]
    $
        \href{http://www.chemspider.com/Chemical-Structure.2497481.html}{\signal{{3OC12}{-}HSL}}
        \TODO{?}
        \protein{LasR}
        \TODO{?}
        \promoter{Las}
    $

\item[\ce{c_3}:]
    $
        \signal{NG}
        \TODO{?}
        \protein{FdeR}
        \TODO{?}
        \promoter{FdeA}
    $

    \TODO{
    what is NG? 
    is it 
    \href{https://en.wikipedia.org/wiki/Naringenin}{Naringenin}?
    cf.~\url{https://pubmed.ncbi.nlm.nih.gov/24188962/}
    }
\end{itemize}


\TODO{tubes and such:}
%
In phase A.
%
Add \ce{w_A} everywhere.
%
Then:
%
\begin{enumerate}
\item 
    Admix the medium of Bit~0 and Bit~1 
    (containing the signals \ce{s_0} and \ce{s_1})
    to Player~A,
    wait for Player~A to respond
    (with \ce{r_0}/\ce{r_1}).
    
\item
    Admix the medium of Player~A
    (with \ce{r_0}/\ce{r_1}/\ce{s_0}/\ce{s_1})
    to Bit~0, wait for response
    (with \ce{c_1}).
    
\item
    Admix the media of Player~A and Bit~0
    (with \ce{r_0}/\ce{r_1}/\ce{s_0}/\ce{s_1}/\ce{c_1})
    to Bit~1, wait for response
    (with \ce{c_2}). 
    
\item
    
\item
    Admix the medium from Bit~3 to Bit~4.
    
\end{enumerate}




\begin{table}[hpbt]
    \centering
    %
    % Table made with
    % https://deepnote.com/project/998c51eb-bb3f-4a0f-88e3-c8b50d4d678a#%2Ft_phases.ipynb
    %
    \begin{tabular}{l||cc|ccc|ccc|ccc|cc}
    {} &      \ce{r_0} &      \ce{r_1} &    \ce{s_0} &      \ce{d_0} &      \ce{c_1} &      \ce{s_1} & \ce{d_1} &      \ce{c_2} &    \ce{s_2} &      \ce{d_2} &      \ce{c_3} &    \ce{s_3} &      \ce{d_3} \\ 
    \hline
    1 &  $\downarrow$ &  $\downarrow$ &  $\uparrow$ &               &               &    $\uparrow$ &          &               &             &               &               &             &               \\ 
    \hline
    2 &    $\uparrow$ &               &  $\uparrow$ &  $\downarrow$ &  $\downarrow$ &               &          &               &             &               &               &             &               \\ 
    \hline
    3 &               &    $\uparrow$ &             &               &    $\uparrow$ &  $\downarrow$ &          &  $\downarrow$ &             &               &               &             &               \\ 
    \hline
    4 &               &               &             &               &               &               &          &    $\uparrow$ &  $\uparrow$ &  $\downarrow$ &  $\downarrow$ &             &               \\ 
    \hline
    5 &               &               &             &               &               &               &          &               &             &               &    $\uparrow$ &  $\uparrow$ &  $\downarrow$ \\ 
    \hline
    \end{tabular}
    %
    %
    \caption{%
        Substeps of each phase A/B.
        Arrows indicate emit ($\uparrow$) and compute ($\downarrow$).
    }
    \label{t:}
\end{table}


\TODO{mention a microfluids setup}

\TODO{how many intercellular signaling pathways do we need and what are they?}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Logical circuits}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\subsection{The players}

\TODO{Table \ref{t:logical-playera}}

In
\cite{NielsenETAL2016}
we find the gates
$
	\ce{r_0} = 
	\mathop{\mathsf{0x0E}}
	(\ce{s_0}, \ce{s_1}, \ce{w_A})
$
and
$
	\ce{r_1} =
	\NOT
	\mathop{\mathsf{0xF6}}
	(\ce{s_0}, \ce{s_1}, \ce{w_A})
$
among those circuits that performed well.
%


%



\input{t_logical-playera}

\subsection{Subtractor} \label{s:sub}

\TODO{Table \ref{t:subtractor1}}





\input{t_subtractor1}


%
Fig.~\ref{f:logical-subtractor01}
illustrates the principle of delayed response
for the half-subtractor,
cf.~Table~\ref{t:subtractor1}.
%


\section{Genetic toolbox}


\TODO{mention 3-input boolean functions \cite{NielsenETAL2016}}

\TODO{Specify initial conditions}


\subsection{Transducers}


\subsubsection*{AmtR}

The TetR family member AmtR is 
the central regulator of nitrogen starvation response
in 
the Gram-positive bacterium
\emph{Corynebacterium glutamicum}
\cite{JakobyETAL2000}.
%
This repressor is released
by
the trimeric adenylylated $\mathrm{P_{II}}$-type 
signal-transduction protein GlnK
\cite{BeckersETAL2005, SevvanaETAL2017},
which is present upon
nitrogen starvation. 
%
The AmtR box and the regulon is 
characterized in 
\cite{BeckersETAL2005},
and
several variants are compared 
in \cite{MuhlETAL2009}.
%
%
\TODO{Do we need (to take care of) the GlnK release mechanism?}
%
It seems clear that 
AmtR sits on the DNA as a dimer
\cite{SevvanaETAL2017},
\cite[\S3.4.2]{Schwab2019}.
%
%
We therefore assume
the mechanism
\begin{subequations}
\[
	\ce{
		2 \protein{AmtR} + \promoter{AmtR}
		& <=>>
		\protein{AmtR}_2 \with \promoter{AmtR}
	}
	%
	\\
	%
	\ce{
		\promoter{AmtR} 
		& ->
		\promoter{AmtR} + \protein{Output}
	}
\]
\end{subequations}
and
the kinetics:
\begin{subequations}
\[
	\label{e:AmtR_Act}
	%
	\promoter{AmtR} 
	& =
	\frac{1}{1 + k_{\ref{e:AmtR_Act}} \protein{AmtR}^{n_{\ref{e:AmtR_Act}}}}
	\promoter{AmtR}_\text{total}
	%
	% here, k = forward / backward
	%
	,
	\\
	%
	\label{e:AmtR_Out}
	%
	\tfrac{\d}{\d{t}}
	\protein{Output} 
	& =
	k_{\ref{e:AmtR_Out}}
	\promoter{AmtR}
	.
\]
\end{subequations}
%
%

% Further literature
%\cite{MuhlETAL2009}
%\cite{BuchingerETAL2009}

\ra{working on this}


\subsubsection*{...}


\subsection{Sensors}

\TODO{Explain intercellular signaling pathways from \cite{DuETAL2020}}

\subsubsection*{3OC6-HSL/LuxR}

\TODO{fix 3OC6}
\TODO{spacing of act and rep}

Pathway:
\[
	\signal{{3OC6}{-}HSL} \act \protein{LuxR} \act \promoter{Lux} \act \protein{Output}
	.
\]

 
The transcription activator \protein{LuxR}
occurs in Gram-negative bacteria
such as \emph{Vibrio fischeri}.
%
The bacterium is permeable to the (auto)inducer,
here \signal{3OC6\text{-}HSL}.
%
The inducer binds to the N-terminal of \protein{LuxR},
which otherwise inhibits its
functional C-terminal \cite{StevensDolanGreenberg1994}.
%
%
The purified C-terminal binds 
upstream of the \emph{lux} box 
(which is centered at $-42.5$bp \cite{EglandGreenberg1999});
however, 
together with the RNA Pol,
it protects the \emph{lux} box and the \emph{lux} operon
promoter
\cite{StevensDolanGreenberg1994}.
%
%
%
We abbreviate
$\protein{LuxR^\star} := \signal{3OC6\text{-}HSL} \with \protein{LuxR}$
and
$\promoter{Lux^\star} := \protein{LuxR^\star} \with \promoter{Lux}$.
%
%\TODO{
%Let $\protein{LuxR^\star}$ denote 
%the activated form
%$\signal{3OC6}:\protein{LuxR}$.
%}
%
%
We assume the mechanism:
%
\begin{subequations}
\[
	\ce{
		\signal{{3OC6}{-}HSL} + \protein{LuxR}
		& <=>>
		\protein{LuxR^\star}
	}
	\\
	\ce{
		\protein{LuxR^\star} + \promoter{Lux}
		& <=>>
		\promoter{Lux^\star}
	}
	\\
	\ce{
		\promoter{Lux^\star}
		& ->
		\promoter{Lux^\star} + \protein{Output}
	}
	.
\]
\end{subequations}
%
%=======
%	\protein{LuxR^\star} + \protein{RNApol} + \promoter{P_{Lux}}
%	& \longleftrightarrow
%	\protein{LuxR^\star} + \protein{RNApol} : \promoter{P_{Lux}}
%	\\
%	\protein{RNApol} : \promoter{P_{Lux}}
%	& \longrightarrow
%	\protein{RNApol} + \promoter{P_{Lux}} + \protein{Output}
%>>>>>>> c8a0f57bdb0a7b80747263819cbb64562e087765
%\TODO{
%\[
%	\signal{3OC6} + \protein{LuxR}
%	& \longleftrightarrow
%	\protein{LuxR^\star}
%	\\
%	\protein{LuxR^\star} + \protein{RNApol} + \promoter{Lux}
%	& \longleftrightarrow
%	\protein{LuxR^\star} + \protein{RNApol} : \promoter{Lux}
%	\\
%	\protein{RNApol} : \promoter{Lux}
%	& \longrightarrow
%	\protein{RNApol} + \promoter{Lux} + \protein{Output}
%\]
%}
% https://chem.libretexts.org/Bookshelves/Biological_Chemistry/Supplemental_Modules_(Biological_Chemistry)/Enzymes/Enzymatic_Kinetics/Sigmoid_Kinetics
%
%
%
and the kinetics:
%
\begin{subequations}
\[
	\label{e:LuxR_Act}
	%
	\protein{LuxR^\star} 
	& =
	\frac{
		\signal{3OC6\text{-}HSL}
	}{
		k_{\ref{e:LuxR_Act}} + \signal{3OC6\text{-}HSL}
	}
	\protein{LuxR}_\text{total}
%	\TEXT{initially}
%	\protein{LuxR^\star} = 0
	,
	%
	\\
	%%
	\label{e:P_Lux_Act}
	%
	\promoter{Lux^\star} 
	& =
	\frac{
		\protein{LuxR^\star}
	}{
		k_{\ref{e:P_Lux_Act}} + \protein{LuxR^\star}
	}
	\promoter{Lux}_\text{total}
%	\TEXT{initially}
%	\promoter{Lux^\star} = 0
	,
	%
	\\
	%
	\label{e:P_Lux_Out}
	%
	\tfrac{\d}{\d{t}}
	\protein{Output}
	& =
	k_{\ref{e:P_Lux_Out}} \promoter{Lux^\star}
	.
\]
\end{subequations}
%
%
In combination, we have
\[
	\tfrac{\d}{\d{t}}
	\protein{Output}
	=
	k_{\ref{e:P_Lux_Out}}
	\promoter{Lux}_\text{total}
	\frac{
		\signal{} \protein{}_\text{total}
	}{
		k_{\ref{e:LuxR_Act}} k_{\ref{e:P_Lux_Act}}
		+
		k_{\ref{e:P_Lux_Act}} \signal{}
		+
		\signal{} \protein{}_\text{total}
	}
	.
\]


\subsubsection*{Salicylate/NahR}

Pathway:
%
\[
	\signal{Sal} \act \protein{NahR} \act \promoter{Sal} \act \protein{Output}
	.
\]

%

According to 
\cite{SchellWender1986, HuangSchell1991},
the transcription activator \protein{NahR}
binds to the recognition site of \promoter{Sal} at
$-83$ to $-45$,
and does so without the inducer \cite[p.10837]{HuangSchell1991}.
The \protein{NahR} hence activate the expression of \promoter{Sal}.
%
%
%
In \cite{SchellBrownRaju1990},
it was suggested 
that the active configuration of \protein{NahR} is a tetramer,
while \cite{ParkLimShin2005}
reported that 
there could be three different complexes
%
$\protein{NahR} \with \promoter{Sal}$.

\TODO{make coherent}

\hh{
Nevertheless, the configuration changes 
then promotes the RNA polymerase binding near $-35$ (upstream of \promoter{Sal}).
%
%
The inducer
(here \signal{Sal}) can induce the conformation change of \protein{NahR},
and activates the promoter \promoter{Sal}.
}

%
%
Following \cite{Peking2013},
we suppose
that 
$4 \times \protein{NahR}$ bind to the DNA,
and
transcription starts
once
one $\signal{Sal}$
binds to each \protein{NahR}.
%
For the sensors we assume an abundance
of \protein{NahR}
\TODO{describe constitutive promoter},
hence the inactive form of the promoter
is $\protein{NahR}_4 \with \promoter{Sal}$.
%
%
We write
$
	\promoter{Sal^\star} :=
	\signal{Sal}_4 \with \protein{NahR}_4 \with \promoter{Sal}
$
for the active form.
%
%
We assume the mechanism
%
\begin{subequations}
\[
	\ce{
		\signal{Sal} + 
		\signal{Sal}_{k - 1} \with \protein{NahR}_4 \with \promoter{Sal}
		& <=>>
		\signal{Sal}_k \with \protein{NahR}_4 \with \promoter{Sal}
	}
	,
	\quad \text{k} = 1, 2, 3, 4,
	%
	%
	\\
	%
	%
	\ce{
		\promoter{Sal^\star}
		&
		->
		\promoter{Sal^\star} + \protein{Output}
	}
\]
\end{subequations}
%
%Mechanism (here \protein{NahR^\star} denotes the activated (tetramer) form):
%\[
%	\signal{Sal} + \protein{NahR}
%	& \longleftrightarrow
%	\protein{NahR^\star}
%	 \\
%	 \protein{NahR^\star} + \protein{RNApol} + \promoter{P_{Sal}}
%	& \longleftrightarrow
%	\protein{NahR^\star} + \protein{RNApol} : \promoter{P_{Sal}}
%	\\
%	\protein{RNApol} : \promoter{P_{Sal}}
%	& \longrightarrow
%	\protein{RNApol} + \promoter{P_{Sal}} + \protein{Output}
%	.
%\]
%
%
which we implement as:
%
\begin{subequations}
\[
	\label{e:NahR_Act}
	%
	\promoter{Sal^\star}
	& =
	\frac{
		\signal{}^{n_{\ref{e:NahR_Act}}}
	}{
		k_{\ref{e:NahR_Act}} + \signal{}^{n_{\ref{e:NahR_Act}}}
	}
	\promoter{Sal}_\text{total}
	,
	%
	%
	\\
	%
	%
	\label{e:NahR_Out}
	%
	\tfrac{\d}{\d{t}}
	\protein{Output}
	& =
	k_{\ref{e:NahR_Out}} 
	\promoter{Sal^\star}
	.
\]
\end{subequations}



\subsubsection*{pC-HSL/rpaR}

Pathway:
%
\[
	\signal{pC{-}HSL} \act \protein{RpaR} \act \promoter{rpa}
	\act 
	\protein{Output}
	.
\]

\hh{
The p-coumaroyl-HSL -- RpaR system 
(\signal{pC\text{-}HSL})
works similarly to AHL/LuxR
\TODO{use `protein`?}.
%
However, as they come from an anoxygenic phototrophic soil bacterium strain, 
the luxIR-type pair, rpaI and rpaR shows good orthogonality to many
QS signals in use \cite{SchaeferETAL2008}.
}

\ra{
The \emph{Rhodopseudomonas palustris} transcriptional regulator
\protein{RpaR},
when purified,
binds an inverted repeat element 
centered at $-48.5$bp
of its promoter
\cite{HirakawaETAL2011}.
%
%
Transcription depends on 
the inducer 
\emph{p}-coumaroyl-homoserine lactone,
or \signal{pC} for short.
%
%
%There is
%pC-HSL-RpaR-activated antisense transcription of \protein{rpaR}.
%
%
The complex 
$\signal{pC} : \protein{RpaR}$
bound to the promoter
activates transcription
\cite[Discussion]{HirakawaETAL2011}.
}


\subsubsection*{XX}



\subsubsection*{XX}

\ra{--}

\subsubsection*{XX}



\subsubsection*{XX}




\subsection{Recombinases}

\subsubsection*{Bxb1}


Bxb1 gp35/gp47
serine-integrase/excisionase
(Int/Xis)
pair
directs the infection cycle of Bxb1 
in
\emph{Mycobacterium smegmatis}.
%
Stoichiometry, start codons,
degradation rates (proteolysis tags)
and DNA substrate copy number 
were optimized 
in \emph{E.~coli}
in \cite{BonnetSubsoontornEndy2012}
to 
assemble a memory bit
that is
stable and responsive over several generations.
%
A promoter of choice is flanked 
by Int recognition sites
and
its directionality is 
\emph{set}
by Int
and 
\emph{reset}
by Int+Xis.
%
Switching times of $\sim$4h were demonstrated
(but the authors expected 30min to be realistic).
%
Unless state otherwise,
we convene that
the \emph{reset} state corresponds to 
positive output (memory bit is \emph{on}).
%
The final construct
\cite[Fig.~4A]{BonnetSubsoontornEndy2012}
consists of an Int gene 
and
a Xis+Int operon
induced by two separate signals,
which facilitates stoichiometry optimization.
%
However, 
in our design,
an external pulse (say, \signal{w_B}) controls Int,
and
an internal compute unit (say, \ce{s_0 \XOR r_0}) controls Xis.
%
The flanked promoter is activated by another
external signal (say, \signal{w_A}).
%
This achieves the separation 
discussed in \S\ref{s:sub} and Fig.~\ref{f:logical-subtractor01}.

%
% SymBiology:
% https://github.com/numpde/ibiocomp/tree/main/code/20201230_Integrase

%




\subsubsection*{XX}


\subsubsection*{XX}


\subsubsection*{XX}


\subsection{Ribocomputing}

\subsubsection*{AND gate}

RNA-only circuits
were proposed and in evaluated in \emph{E.~coli} in \cite{GreenETAL2017}.
%
A toehold switch for AND logic
\cite[\href{https://www.nature.com/articles/nature23271/figures/2}{Fig.~2e}]{GreenETAL2017}



\TODO{AND gate from \cite{GreenETAL2017}}
\ra{maybe not necessary, since we have another AND-mechanism
with the dimer}


\subsection{3-input XOR gate}


In \cite[\href{https://bmcbiol.biomedcentral.com/articles/10.1186/s12915-015-0146-0/figures/5}{Fig.~5}]{Wong2015}, 
a \XOR gate is built with two plasmids. 
%
On the first plasmid, 
a constitutive promoter \TODO{need to specify at some point}
promotes the synthesis of proteins AraC and RhaS.
%
% Ara
\ra{
AraC is an arabinose-induced transcription activator
of the promoter pBAD upstream of HrpS. 
%
% Rha
Similarly, RhaS activates the promoter pRHAB in the presence of inducer rhamnose and then synthesis of HrpR.
%
HrpR and HrpS form a heterodimer HrpRS
that binds to and activates the promoter pHrpL 
upstream of the repressor $\lambda$CI 
on the second plasmid.
%
At the same time, inducers arabinose and rhamnose 
each promote the expression
of the reporter RFP on the second plasmid via
the promoters pBAD and pRHAB,
which are, however,
modified to allow $\lambda$CI to bind
and prevent expression.
%
In this way, 
RFP is expressed when either one of them is present but not both. 

It is noted \TODO{by whom, where \hh{in this exact cited paper, when they discuss why the copy number is like this} 
\ra{so what are the numbers? if you make a specific claim, provide a specific reference}
}
that the promoter pHrpL is relatively weak.
%
The authors propose to 
increase the copy number of 
the second plasmid (where the promoter is located)
in order to
increase the dynamic range of the activator dimer.
}

%
\ra{
We could not find
$
	\ce{d_1} 
	= 
	\mathop{\mathsf{0x69}}
	(\ce{s_1}, \ce{r_1}, \ce{c_1})
	=
	\NOT
	\mathop{\mathsf{0x96}}
	(\ce{s_1}, \ce{r_1}, \ce{c_1})
$,
i.e.~the 3-\XOR,
cf.~Table~\ref{t:subtractor1},
explicitly in \cite{NielsenETAL2016}.
%
}
%
\ra{
To implement this,
we cascade
two constructs analogous to \cite{Wong2015}
using two orthogonal heterodimers.
}

\TODO{
we only have one 3-input XOR gate, therefore ABC should be specified
somewhere here
as what they are, ie: 
\ce{s_1}/\ce{r_1}/\ce{c_1} mapping to \ce{d_1}.
%
however, which should be which? --
how about 
\ce{c_1} = C.
}

\TODO{
\ra{urgent}
what are the actual signaling molecules we're using?
we should settle this asap, because now everything depends on it,
including the symbio;
thus, first we need the ``experimental design'' section
\hh{Ara, Rha and IPTG is what I suggested for the 2 layered xor gates.}}

\begin{figure}
    \centering
    \includegraphics{images/circuits/3_input_xor.png}
    \caption{3 input \XOR logic gate as cascading \XOR gates}
    \label{f:my_label}
    \ra{cf. new figure \ref{f:logical-subtractor01}}
\end{figure}

\hh{
The first 2-\XOR (A and B) use the above implementation 
(i.e.~with heterodimer HrpRS) to regulate the expression of pchB. 
%
The 3rd inducer C then triggers the expression of pchA.
%
Similarly, only when pchB and pchA are co-expressed, the dimer pchBA can trigger the synthesis of salicylic acid, which then regulates the activator NahR to activate the expression of repressor McbR \cite{Rey2005}.

}

\begin{figure}
    \centering
    \includegraphics[width = \linewidth]{images/gene_circuits_xor/P1.png}
    \includegraphics[width = \linewidth]{images/gene_circuits_xor/P2.png}
    \includegraphics[width = \linewidth]{images/gene_circuits_xor/P3.png}
    \includegraphics[width = \linewidth]{images/gene_circuits_xor/P4.png}
    \caption{Construct of layered \XOR gate. Inducer A, B and C are Ara, Rha and IPTG, which regulate the activity of pBAD, pRHAB and pTAC respectively. Here the tagged promoters refers to the ones that can be blocked by the upstream repressor. \hh{This is just a design sketch taht i came up with and drew with \url{https://sboldesigner.github.io}. For the final report we would use something looked nicer for the design diagram}
    \ra{if you keep them online please provide a url and/or upload the relevant files (not just the figures)}
    }
    \label{f:gene_circuit_xor}
\end{figure}


\TODO{
what is the url of the circuit figure, if any; why is it upside-down
\hh{there is no url of the circuit figure. it is just a sketch indicating 2 layered xor gate. and I dont know if there is a rule for which way should be up and which way is down}
\ra{cf. new figure \ref{f:logical-subtractor01}}
}

\ra{we don't do PNG b/c that doesn't scale well (neither up nor down). unfortunately, www.circuit-diagram.org doesn't export eps or pdf, but it does export svg, from which I make pdf [can delete this]}

\subsubsection*{Cascading-OR XOR gate}

\TODO{other XOR gates (perhaps briefly) one might consider}


The gates
$
	\ce{c_2} 
%	=
%	\mathop{\mathsf{0x4D}}
%	(\ce{r_1}, \ce{s_1}, \ce{c_1}) 
	=
	\mathop{\mathsf{0x4D}}
	(\ce{c_1}, \ce{s_1}, \ce{r_1}) 
%	=
%	\NOT
%	\mathop{\mathsf{0x8E}}
%	(\ce{r_1}, \ce{c_1}, \ce{s_1})
	=
	\NOT
	\mathop{\mathsf{0x8E}}
	(\ce{c_1}, \ce{r_1}, \ce{s_1})
$,
which are
symmetric in \ce{r_1}/\ce{c_1},
were successfully
implemented in \cite{NielsenETAL2016}.
%
%


\subsubsection*{3x recombinase}


\TODO{
\ra{Issure:} Can we have a resettable \XOR with recombinases
eg as theorized here in Fig.~2g?
\url{https://www.nature.com/articles/s41598-017-07386-3/figures/2}
}


\ra{other notes:}

\ra{2-input \XOR with recombinase \cite{BonnetETAL2013}}
\ra{theoretical 3-input \XOR with recombinase \cite{ChiuJiang2017}}

\ra{full adder in ``mammalian'' cells \cite{WeinbergETAL2017} (\url{https://www.addgene.org/87552/}),
communicating \cite{AuslaenderETAL2017}}

%






\section{Simulations}

\subsection{PlayerA}

\url{https://github.com/numpde/ibiocomp/tree/main/code/20201231_SymBio_All/PlayerA}

\TODO{Some details missing, also need to carefully choose the wake-up signal \ce{w_A} / \ce{w_B}:
\href{https://drive.google.com/file/d/1X1uEVNkUiXZqNz7C7m36YjW-g1XctIH0}{click for figure (ignore black part)}}

\subsection{Bit 0}

\url{https://github.com/numpde/ibiocomp/tree/main/code/20201231_SymBio_All/Bit0}

\TODO{this could have the same problem as described in Bit 1}

\subsection{Bit 1}

\url{https://github.com/numpde/ibiocomp/tree/main/code/20201231_SymBio_All/Bit1}


\TODO{\ra{Issue: how to implement this ``Dock'' (and a better name)?}}
\ra{%
Note: This seems similar to
\cite[\href{{https://www.pnas.org/content/pnas/104/8/2643/F1.large.jpg?width=800&height=600&carousel=1}}{Fig.~1}]{WeberETAL2007},
also
\cite{NilgiriwalaETAL2014}.
}
%
Consider bit 1 of Subtractor A.
%
Suppose \ce{s_1 r_1 c_1 = 101},
in which case \ce{c_2 = 0}.
%
Suppose a transcription factor \ce{Y},
which encodes \ce{c_2},
promotes expression of \ce{Xis} in the memory subunit.
%
%
When \ce{w_B} is up,
access to memory is on.
%
Since we want \ce{c_2 = 0},
the amount of \ce{Y} is low,
but
around the timepoint when 
\ce{w_B} is turned off
and the inputs begin to change,
\ce{Y} can experience a transient bump.
%
This is sufficient to produce \ce{Xis}
and thus to cause an unintended \emph{reset}
towards \ce{c_2 = 1}.
%
We found two ways to mitigate this
in the computational model.
%
One is to make expression of \ce{Xis}
conditional on \ce{w_B} also
but this requires that \ce{w_B} goes down
before other change
and could be sensitive to degradation rates.
%
A more robust way
is to introduce a buffer for \ce{Y},
which we call ``Dock'',
that 
%
i)
has high affinity for \ce{Y};
%
ii)
sequesters a large amount of \ce{Y},
about 30x compared to the 
\href{https://en.wikipedia.org/wiki/EC50}{EC50}
of \ce{Y} \act \ce{Xis};
%
iii)
releases \ce{Y} for degradation when \ce{w_A} is up.
%
In this way, 
a transient bump in \ce{Y} is absorbed 
by the Dock,
but a sustained signal 
(while \ce{w_B} is up)
will overflow the Dock and activate 
transcription of \ce{Xis}.
%


%

\section{Discussion}

\TODO{risks}


\footnotesize
\bibliographystyle{apalike}
\bibliography{refs}


\section*{Appendix}

\TODO{On Figure \ref{f:logical-subtractor01}.}

% https://www.circuit-diagram.org/editor/c/075f8cf9d084400b94f1384cc1c3ba96
\begin{figure}[phbt]
\centering
\includegraphics[width=0.9\linewidth]{circuits/Logical-HalfSubtractor0.svg.pdf}
\caption{%
A logic circuit representation of Subtractor~A:
Bit~0 and Bit~1.
%
The result of the compute unit is written to the memory unit
(bistable switch)
when the signal \ce{w_B} is present.
%
The value of the memory unit is
forwarded to the output
when the signal \ce{w_A} is present.
}
\label{f:logical-subtractor01}
\end{figure}




\clearpage

\SHOWTODOS




\leavevmode\vfill{\tiny\color{lightgray}\hfill{\DTMnow}}
\end{document}





