\documentclass[12pt,notitlepage]{article}

% Overleaf project:
% https://www.overleaf.com/project/5fdb880ef955c3369964d601

% View-only link:
% https://www.overleaf.com/read/hkybwqtbtvjb

\usepackage{amsmath, amsfonts, amssymb}
\usepackage[svgnames]{xcolor}
\usepackage{datetime2}
\usepackage[
	colorlinks=true, 
	citecolor={DarkRed}, urlcolor={DarkBlue}, linkcolor={DarkBlue},
]{hyperref}


% 
\usepackage[version=4]{mhchem}

\usepackage{fullpage}

% Paragraph spacing
\usepackage{parskip}

\usepackage{xspace}

\usepackage{graphicx}
\graphicspath{{../code/20201231_SymBio_All/}{../images/}}
\DeclareGraphicsExtensions{.pdf,.eps,.png}

% https://tex.stackexchange.com/questions/202046/width-of-the-caption-of-a-figure
% https://tex.stackexchange.com/questions/29039/how-to-limit-the-figure-caption-width
\usepackage[margin=20px]{caption}

% https://tex.stackexchange.com/questions/20792/how-to-superimpose-latex-on-a-picture
\usepackage[percent]{overpic}

%\usepackage{epstopdf}

% Tables (the order matters here)
\usepackage{makecell}
\usepackage{booktabs}
\usepackage{arydshln}

% https://en.wikibooks.org/wiki/LaTeX/Footnotes_and_Margin_Notes#Margin_Notes
\usepackage{marginnote}

% For editing purposes:
%\usepackage[margin=10pt]{geometry}

% https://latex.org/forum/viewtopic.php?t=10456
\usepackage{titlesec}
\titleformat{\subsubsection}[runin]% runin puts it in the same paragraph
{\normalfont\bfseries}% formatting commands to apply to the whole heading
{\thesubsubsection}% the label and number
{}% space between label/number and subsection title
{}% formatting commands applied just to subsection title
[.]% punctuation or other commands following subsection title


\newcommand{\TODO}[1]{\textrm{\color{red}TODO: #1}}
\input{todomech}


\renewcommand{\d}{\mathrm{d}}

\newcommand{\NOT}{\ensuremath{\mathop{\mathsf{not}}}\xspace}
\newcommand{\AND}{\ensuremath{\mathop{\mathsf{and}}}\xspace}
\newcommand{\OR}{\ensuremath{\mathop{\mathsf{or}}}\xspace}
\newcommand{\XOR}{\ensuremath{\mathop{\mathsf{xor}}}\xspace}

\newcommand{\TEXT}[1]{\quad\text{#1}\quad}
\newcommand{\with}{\text{$\,{:}\,$}}

\newcommand{\cbra}[1]{{\ensuremath{\color{gray}{#1}}}}
\newcommand{\signal}[1]{{{\cbra{\langle}\ce{#1}\cbra{\rangle}}}}
\newcommand{\protein}[1]{{{\cbra{(}\ce{#1}\cbra{)}}}}
\newcommand{\promoter}[1]{{{\cbra{[}\ce{#1}\cbra{]}}}}

% https://tex.stackexchange.com/questions/543953/arrow-with-blunted-end-head-in-math-mode
\newcommand{\act}{\ {\ensuremath{\mathbin{\to}}}\ }
\newcommand{\rep}{\ {\ensuremath{\mathrel{\raisebox{-.3ex}{\rotatebox{90}{\scalebox{1}[1.2]{$\bot$}}}}}}\ }

\def\[#1\]{\begin{align}#1\end{align}}

% https://tex.stackexchange.com/questions/114113/how-to-label-text-with-equation-number
\newcommand{\eqnum}{\leavevmode\hfill\refstepcounter{equation}\textup{{(\theequation)}}}

\newcommand{\starlink}[1]{\textsuperscript{\makebox[0pt]{\href{#1}{\color{white}$\star$}}}}

\newcommand{\hh}[1]{{\color{Purple}#1}}
\newcommand{\ra}[1]{{\color{Blue}#1}}


\title{ibiocomp project}
\author{RA \& HH}
\date{\today}


\begin{document}

\maketitle

% \ra{LOCKED 'COS WORKING WITH ANOTHER COPY (WRITE EMAIL IF NEEDED)}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Introduction}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\subsection{The game of 15 sticks}

We design genetic circuits
that play the following game:
%
\begin{quote}
	Two players start with 15 sticks
	and alternatingly 
	take 1, 2 or 3 sticks.
	\\
	The last player to take a stick loses.
\end{quote}

Because a player can complement
the number of sticks taken by the opponent,
there is a deterministic winning strategy
of the first-mover
as shown in Table \ref{t:logical-playera}.
%
%
We implement this strategy for both players.


\subsection{General design} \label{s:general}

We encode in binary 
the number of sticks left as
\[
    \label{e:s0123}
    %
    \text{number of sticks left} = \ce{8 s_3 + 4 s_2 + 2 s_1 + s_0}
    .
\]
%
%
By the 4-periodicity of the winning strategy,
only the two lowest bits \ce{s_1}/\ce{s_0}
matter to the player.
%
%
The response of a player is a number 1, 2, or 3,
which we encode in binary as
\[
    \label{e:r01}
    %
    \text{number of sticks taken by a player} = \ce{2 r_1 + r_0}
    .
\]
%
The bits in \eqref{e:s0123}--\eqref{e:r01}
are transmitted by 
intercellular signaling molecules,
cf.~\S\ref{ss:experiment}.
%
Although our players are essentially identical,
we model them separately and with one difference:
Player A (B) becomes active when 
the extracellular signal \ce{w_A} (\ce{w_B}) is supplied.

%

The \emph{subtractor} module
keeps track of the number of sticks left \eqref{e:s0123}.
To appreciate the key design issue, suppose
we manually transmit the response \ce{r_1 r_0 = 01}
of a player
into the medium of the subtractor.
It should compute the new number of the sticks,
yet, it has to do so once
and not keep subtracting the value \ce{r_1 r_0}
that is still present.
One solution is single-use subtractors
but then the result of each computation
has to be imparted to an unused subtractor.
Barring excessive manual intervention,
a feasible design
seems to require some separation of
reading/storing the current state,
computing the new state and communicating it
--
temporally or spatially.
For our purposes
this may include
modest parallelism in intercellular signaling
(but not frequency encoding).
%
%
We propose 
a subtractor composed of two parts A and B,
and
a workflow 
%of Phase A, Interphase, Phase B, Interphase, etc.,
as shown in Table \ref{t:workflow}.
%
%
%
Thus,
in Phase A characterized by the supply of 
the master signal \ce{w_A},
Subtractor A
emits the current state \ce{s},
Player A responds with \ce{r},
while
Subtractor B
reads \ce{s}/\ce{r}
and
computes the new state % \ce{s} 
{silently}
and remembers the result
in order to emit it in the subsequent Phase B.
%
The medium is cleared, before
Phase B is initiated by
supplying the master signal \ce{w_B}.
%
%
In short,
we require
a gated memory module and a delayed response,
cf.~\S\ref{ss:sub}.

%

Refinements could include
a)
dealing with
a player who might attempt
to take more sticks than \eqref{e:s0123};
b)
dedicated sensor cells 
that report 
on the presence of signaling molecules 
in order to 
monitor the state of the game.

%

%%% TABLE %%%
\input{t_workflow}



\subsection{Logic circuits}


\subsubsection*{The players} \label{ss:players}

Both players
follow the same deterministic strategy,
which is a winning strategy for the first-mover.
%
They are identical except
that player A responds when \ce{w_A} is present
and player B responds upon \ce{w_B}.
%
The logic is shown in Table \ref{t:logical-playera}.

%%% TABLE %%%
\input{t_logical-playera}


\subsubsection*{Subtractor} \label{ss:sub}

The subtractor senses the current number of sticks and the response from the player. It is composed of 4 computing units, denoted by Bit 0, Bit 1, Bit 2 and Bit 3. The subtraction is done bit by bit. Bit 0 senses the current \ce{s_0} and \ce{r_0}, and computes the difference  \ce{d_0} and the carry flag \ce{c_1}. The carry flag is then passed on to the next computing unit, i.e Bit 1 in this case. Bit 1 subtracts \ce{r_1} and \ce{c_1} from \ce{s_1} (\ref{t:subtractor1}, middle column). The carry flag \ce{c_2} computed from Bit 1 is passed on to the next computing unite Bit 2, which subtracts \ce{c_2} from \ce{s_2}. Therefore, for the computing unit Bit 0, Bit 2 and Bit 3, only half-subtractor is required (2 inputs) while the Bit 1 requires a full subtractor (3 inputs).


%%% TABLE %%%
\input{t_subtractor1}

%
Fig.~\ref{f:logical-subtractor01}
illustrates the principle of delayed response
for the half-subtractor,
cf.~Table~\ref{t:subtractor1}.








\subsection{Experimental design} \label{ss:experiment}

We distinguish
\emph{extracellular} chemical signals 
that are supplied by the experimenter
(the master signals \ce{w_A}/\ce{w_B})
and
\emph{intercellular} signals
that are produced and sensed by cells.
%
%
%
For an experiment in a single test tube 
we need
orthogonal intercellular channels:
\emph{three} for \eqref{e:s0123}
($s_3$ has no downstream),
\emph{two} for \eqref{e:r01},
and
\emph{three} for the carry flags
(see \S\ref{ss:sub}).
%
One channel could be spared 
at the expense
of a more involved computation of 
the last useful carry flag $c_3$.
%
Thus we need \emph{eight} orthogonal intercellular channels,
cf.~Table \ref{t:substeps}.
%
In \cite{DuETAL2020},
ten de novo intercellular signaling pathways
were designed
and
four known QS pathways
were optimized,
however,
only \emph{eight} worked well orthogonally
on the sensor level 
and 
seven on the promoter level
\cite[\href{https://www.nature.com/articles/s41467-020-17993-w/figures/3}{Fig.~3c/g}]{DuETAL2020}.
%
We have tried to assign the signals
to further minimize crosstalk,
in particular 
avoiding LuxR and RpaR in one cell
as warned in \cite[p.6]{DuETAL2020}.
%
The signals are listed in
Table~\ref{t:signals}.
%
%
%
\ra{
As shown in Table \ref{t:signals},
we encode \ce{w_A} by L-arabinose,
\ce{w_B} by IPTG, etc.
%
Our convention is \ce{w_A = 1}
if and only if
the corresponding messenger molecule L-arabinose 
is present in the medium or in the cell 
in sufficiently high quantity.
%
We write \ce{\#w_A}
for
the \emph{number} of L-arabinose molecules 
in the numerical simulations.
%
Ditto for the other signals.
}
%
%
\ra{
Subsequently, we abbreviate
\signal{{3OC6}{-}HSL} by \signal{{3OC6}}, etc.
}
%


%%% TABLE %%%
\input{t_signals.tex}


%

\ra{
To facilitate debugging
we partition the workflow
with spatial compartments
that require manual transmission of 
the medium.
%
Unfortunately, this does not automatically reduce
the number of intercellular channels needed.
}


\TODO{what species are we working with}


\TODO{experimental protocol:}
%
In phase A.
%
Add \ce{w_A} everywhere.
%
Then:
%
\begin{enumerate}
\item 
    Admix the medium of Bit~0 and Bit~1 
    (containing the signals \ce{s_0} and \ce{s_1})
    to Player~A,
    wait for Player~A to respond
    (with \ce{r_0}/\ce{r_1}).
    
\item
    Admix the medium of Player~A
    (with \ce{r_0}/\ce{r_1}/\ce{s_0}/\ce{s_1})
    to Bit~0, wait for response
    (with \ce{c_1}).
    
\item
    Admix the media of Player~A and Bit~0
    (with \ce{r_0}/\ce{r_1}/\ce{s_0}/\ce{s_1}/\ce{c_1})
    to Bit~1, wait for response
    (with \ce{c_2}). 
    
\item
    
\item
    Admix the medium from Bit~3 to Bit~4.
    
\end{enumerate}




\begin{table}[hpbt]
    \centering
    %
    % Table made with
    % https://deepnote.com/project/998c51eb-bb3f-4a0f-88e3-c8b50d4d678a#%2Ft_phases.ipynb
    %
    \begin{tabular}{l||cc|ccc|ccc|ccc|cc}
    {} &      \ce{r_0} &      \ce{r_1} &    \ce{s_0} &      \ce{d_0} &      \ce{c_1} &      \ce{s_1} & \ce{d_1} &      \ce{c_2} &    \ce{s_2} &      \ce{d_2} &      \ce{c_3} &    \ce{s_3} &      \ce{d_3} \\ 
    \hline
    1 &  $\downarrow$ &  $\downarrow$ &  $\uparrow$ &               &               &    $\uparrow$ &          &               &             &               &               &             &               \\ 
    \hline
    2 &    $\uparrow$ &               &  $\uparrow$ &  $\downarrow$ &  $\downarrow$ &               &          &               &             &               &               &             &               \\ 
    \hline
    3 &               &    $\uparrow$ &             &               &    $\uparrow$ &  $\downarrow$ &          &  $\downarrow$ &             &               &               &             &               \\ 
    \hline
    4 &               &               &             &               &               &               &          &    $\uparrow$ &  $\uparrow$ &  $\downarrow$ &  $\downarrow$ &             &               \\ 
    \hline
    5 &               &               &             &               &               &               &          &               &             &               &    $\uparrow$ &  $\uparrow$ &  $\downarrow$ \\ 
    \hline
    \end{tabular}
    %
    %
    \caption{%
        Substeps of each phase A/B.
        Arrows: emit ($\uparrow$) and compute ($\downarrow$).
    }
    \label{t:substeps}
\end{table}


\TODO{mention a microfluids setup?}


\TODO{Specify initial conditions}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Genetic toolbox} \label{s:genetic}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\subsection{3-input gates by Cello} \label{ss:cello}

In
\cite{NielsenETAL2016}
we found the following 
useful three-input one-output gates
among those that performed well:
%
\begin{itemize}
\item
    $
    	\ce{r_0} = 
    	\mathop{\mathsf{0x0E}}
    	(\ce{s_0}, \ce{s_1}, \ce{w_A})
    $,
\item
    $
    	\ce{r_1} =
    	\NOT
    	\mathop{\mathsf{0xF6}}
    	(\ce{s_0}, \ce{s_1}, \ce{w_A})
    $,
\item
    $
    	\ce{c_2} 
    %	=
    %	\mathop{\mathsf{0x4D}}
    %	(\ce{r_1}, \ce{s_1}, \ce{c_1}) 
    %	=
    %	\NOT
    %	\mathop{\mathsf{0x8E}}
    %	(\ce{r_1}, \ce{c_1}, \ce{s_1})
    	=
    	\NOT
    	\mathop{\mathsf{0x8E}}
    	(\ce{c_1}, \ce{r_1}, \ce{s_1})
    $.
    %
    \hfill
    (also equal
    $
       	\mathop{\mathsf{0x4D}}
       	(\ce{c_1}, \ce{s_1}, \ce{r_1}) 
    $
    and
    symmetric in \ce{r_1}/\ce{c_1})
\end{itemize}


We implemented \ce{r_0} mutatis mutandis,
replacing the sensors 
and
replacing the repressor PhlF 
(which clashes with the \ce{s_0} sensor) 
by HlyIIR.
%
%
We found that \ce{r_1} can reuse 
some of the circuit


%
\TODO{explain}





\subsection{Transducers}


\subsubsection*{AmtR}

The TetR family member AmtR is 
the central regulator of nitrogen starvation response
in 
the Gram-positive 
\emph{C.~glutamicum}%
\starlink{https://en.wikipedia.org/wiki/Corynebacterium_glutamicum}
\cite{JakobyETAL2000}.
%
This repressor is released
by
the trimeric adenylylated $\mathrm{P_{II}}$-type 
signal-transduction protein GlnK
\cite{BeckersETAL2005, SevvanaETAL2017},
which is present upon
nitrogen starvation. 
%
The AmtR box and the regulon is 
characterized in 
\cite{BeckersETAL2005},
and
several variants are compared 
in \cite{MuhlETAL2009}.
%
%
\TODO{Do we need (to take care of) the GlnK release mechanism?}
%
It seems clear that 
AmtR sits on the DNA as a dimer
\cite{SevvanaETAL2017},
\cite[\S3.4.2]{Schwab2019}.

% Further literature
%\cite{MuhlETAL2009}
%\cite{BuchingerETAL2009}

\subsubsection*{...}

\TODO{continue}


\subsection{Extracellular signaling} \label{ss:wAB}


The two master signals \ce{w_A} and \ce{w_B}
are encoded by L-arabinose and IPTG,
respectively,
and are supplied by the experimenter.


\ra{Note, from \cite[SM, VII.M]{NielsenETAL2016}:
\begin{quote}
    The sensors used include a truncated AraC 
    (AraC-C280*, referred to as AraC* in this work)
    that has reduced cross-talk with IPTG(40) and its output promoter PBAD
    which is induced using L-arabinose; 
    LacI and its output promoter with a symmetric lac-operator PTac which
    is IPTG-inducible(41); 
    and TetR with its output promoter PTet(42)
\end{quote}
}

\TODO{write about Ara and IPTG}



\subsection{Intercellular pathways}

\TODO{references to \cite{DuETAL2020}}

\TODO{comment on toxicity, see SM of Du}

\TODO{WHAT ABOUT THE SENDERS?}


\subsubsection*{3OC6} \label{ss:3OC6}
 
The transcription activator \protein{LuxR}
occurs in Gram-negative bacteria
such as \emph{Vibrio fischeri}.
%
The bacterium is permeable to the (auto)inducer,
here \signal{{3OC6}}.
%
The inducer binds to the N-terminal of \protein{LuxR},
which otherwise inhibits its
functional C-terminal \cite{StevensDolanGreenberg1994}.
%
%
The purified C-terminal binds 
upstream of the \emph{lux} box 
(which is centered at $-42.5$bp \cite{EglandGreenberg1999});
however, 
together with the RNA Pol,
it protects the \emph{lux} box and the \emph{lux} operon
promoter
\cite{StevensDolanGreenberg1994}.
%
\TODO{finish}


\subsubsection*{IV} \label{ss:IV}

\TODO{describe}

\subsubsection*{DAPG} \label{ss:DAPG}

\TODO{describe}

\subsubsection*{Sal} \label{ss:Sal}

According to 
\cite{SchellWender1986, HuangSchell1991},
the transcription activator \protein{NahR}
binds to the recognition site of \promoter{Sal} at
$-83$ to $-45$
without the inducer \cite[p.10837]{HuangSchell1991}.
%
%
%
In \cite{SchellBrownRaju1990},
it was suggested 
that the active configuration of \protein{NahR} is a tetramer,
while \cite{ParkLimShin2005}
reported that 
there could be three different complexes
%
$\protein{NahR} \with \promoter{Sal}$.

\TODO{make coherent}

\hh{
Nevertheless, the configuration changes 
then promotes the RNA polymerase binding near $-35$ (upstream of \promoter{Sal}).
\TODO{reference?}
%
%
The inducer
(here \signal{Sal}) can induce the conformation change of \protein{NahR},
and activates the promoter \promoter{Sal}.
}

%
%
Following \cite{Peking2013},
we suppose
that 
$4 \times \protein{NahR}$ bind to the DNA,
and
transcription starts
once a $\signal{Sal}$
binds to each \protein{NahR}.
%
For the sensors we assume an abundance
of \protein{NahR}
\TODO{describe constitutive promoter},
hence the inactive form of the promoter
is $\protein{NahR}_4 \with \promoter{Sal}$.
%
%
\TODO{finish}


\subsubsection*{pC} \label{ss:pC}

\hh{
The p-coumaroyl-HSL/RpaR system 
(pC-HSL)
works similarly to AHL/LuxR
\TODO{what is AHL?}
\TODO{use `protein`?}.
%
However,
as they come from an anoxygenic phototrophic soil bacterium strain, 
the luxIR-type pair, rpaI and rpaR shows good orthogonality to many
QS signals in use \cite{SchaeferETAL2008}.
}

\ra{
The \emph{R.~palustris}%
\starlink{https://en.wikipedia.org/wiki/Rhodopseudomonas_palustris}
%
transcriptional regulator
\protein{RpaR},
when purified,
binds an inverted repeat element 
centered at $-48.5$bp
of its promoter
\cite{HirakawaETAL2011}.
%
%
Transcription depends on 
the inducer 
\signal{pC}
(\emph{p}-coumaroyl-homoserine lactone).
%
%
%There is
%pC-HSL-RpaR-activated antisense transcription of \protein{rpaR}.
%
%
The complex 
$\signal{pC} : \protein{RpaR}$
bound to the promoter
activates transcription
\cite[Discussion]{HirakawaETAL2011}.
}
%
%
\TODO{finish}


\subsubsection*{MMF} \label{ss:MMF}

\TODO{describe}


\subsubsection*{3OC12} \label{ss:3OC12}

\TODO{describe}

\url{https://pubmed.ncbi.nlm.nih.gov/12169617/}


\subsubsection*{NG} \label{ss:NG}

\TODO{describe}



\subsection{Memory module} \label{ss:memory}

As explained in \S\ref{s:general}
(cf.~Table~\ref{t:workflow}/\ref{t:subtractor1} and Fig.~\ref{f:logical-subtractor01}),
we need a memory module 
to record and recall the result of the subtraction.
%
To that end we use a ``reversible'' recombinase.
%
%
Specifically,
the Bxb1 gp35/gp47
serine-integrase/excisionase
(Int/Xis)
direct the infection cycle of Bxb1 
in
\emph{M.~smegmatis}%
\starlink{https://en.wikipedia.org/wiki/Mycobacterium_smegmatis}.
%
Stoichiometry, start codons,
degradation rates (proteolysis tags)
and DNA substrate copy number 
were optimized 
in \emph{E.~coli}
in \cite{BonnetSubsoontornEndy2012}
to 
assemble a memory bit
that is
stable and responsive over several generations.
%
A promoter of choice is flanked 
by Int recognition sites
and
its directionality is 
\emph{set}
by Int
and 
\emph{reset}
by Int+Xis.
%
Switching times of $\sim$4h were demonstrated
but the authors expected 30min to be realistic.
%
Their final construct
\cite[Fig.~4A]{BonnetSubsoontornEndy2012}
consists of an Int gene 
and
a Xis+Int operon
induced by two separate signals,
which facilitates stoichiometry optimization.
%
However, 
in our design,
an external pulse (say, \signal{w_B}) controls Int,
and
an internal compute unit (say, \ce{s_0 \XOR r_0}) controls Xis.
%
The \emph{reset} state corresponds to 
positive output, i.e.~the memory bit is \emph{on}.
%
%
The flanked promoter is activated by another
external signal (say, \signal{w_A}).


\subsection{Enhanced AND gate}

\subsubsection*{pchBA}

\TODO{Describe, maybe}

\ra{note: this is used in \cite{DuETAL2020} for Sal}


\subsubsection*{HrpR/HrpS} \label{ss:HrpRS}

\cite[\href{https://www.nature.com/articles/ncomms1516/figures/1}{Fig.~1}]{WangKitneyJolyBuck2011}
proposed 
an \AND gate
in \emph{E.~coli}
using the heterodimeric transcription factor 
$\ce{HrpR} \with \ce{HrpS}$
from 
\emph{P.~syringae}%
\starlink{https://en.wikipedia.org/wiki/Pseudomonas_syringae}%
.
%
%The authors also estimated the parameters in 
%the Hill transfer function.
%
The dimer 
binds upstream of the promoter \ce{hrpL} 
and opens the $\ce{\sigma^{54}{-}RNAP{-}hrpL}$ transcription complex.

\subsubsection*{Ribocomputing}

\cite[\href{https://www.nature.com/articles/nature23271/figures/2}{Fig.~2e}]{GreenETAL2017}
combined
two short input/sensor RNAs
into a ``trigger'' RNA
which opens an RBS-occluding stemloop
on the output mRNA.
%
%
A symmetric variation of the idea is this:
%
Let inputs \ce{A} and \ce{B} 
produce
output \ce{mRNA_A} and \ce{mRNA_B}, resp.,
whose RBS is auto-occluded
by their own 5' tail.
%
Let \ce{A} 
also produce a short RNA
\ce{R^*_B}
such that
the 5' tail of \ce{mRNA_B} preferentially binds \ce{R^*_B},
and vice versa.

\subsubsection*{Cross-inhibition}

Even in our idealized simulations
such \AND gates produce undesired leakage
within larger circuits.
%
%
%
To fix this, we propose that
\emph{low} input \ce{A} 
{inhibit} the downstream of \ce{B},
and vice versa,
see \S\ref{s:x-inh}.




\subsection{Ribocomputing}

\ra{}


\subsection{3-input \XOR}

\ra{NEW TEXT:}
%
%
The subtractor result bit
(cf.~Table~\ref{t:subtractor1})
\[
	\ce{d_1} 
	= 
	\mathop{\mathsf{0x69}}
	(\ce{s_1}, \ce{r_1}, \ce{c_1})
	=
	\NOT
	\mathop{\mathsf{0x96}}
	(\ce{s_1}, \ce{r_1}, \ce{c_1})
	=
	\text{3-\XOR}
	(\ce{s_1}, \ce{r_1}, \ce{c_1})
	=
	((\ce{s_1} + \ce{r_1} + \ce{c_1}) \mathbin{\mathrm{mod}} 2)
\]
is particularly difficult
because the result flips 
whenever any input is flipped.
%
We could not find it explicitly in \cite{NielsenETAL2016}.
%
%
Observe that 3-\XOR is a concatenation
of two \XOR gates (cf.~Fig.~\ref{f:logical-subtractor01})
and 
has the DNF%
\starlink{https://en.wikipedia.org/wiki/Disjunctive_normal_form}:
$
	(\ce{s_1} \wedge \ce{r_1} \wedge \ce{c_1})
	\vee
	(s_1 \wedge \ce{\overline{r_1}} \wedge \ce{\overline{c_1}})
	\vee
	(\ce{\overline{s_1}} \wedge \ce{r_1} \wedge \ce{\overline{c_1}})
	\vee
	(\ce{\overline{s_1}} \wedge \ce{\overline{r_1}} \wedge \ce{c_1})
$.
%
We considered several implementations:
%
\begin{itemize}
\item
	Cascading 
	two \XOR gates
	built on tandem promoters as in
	\cite[\href{https://science.sciencemag.org/content/sci/352/6281/aac7341/F4.large.jpg}{Fig.~3}]{NielsenETAL2016}.
\item
	Cascading
	two \XOR gates, 
	each as in 
	\cite[\href{https://bmcbiol.biomedcentral.com/articles/10.1186/s12915-015-0146-0/figures/5}{Fig.~5}]{Wong2015}
	but
	with two orthogonal heterodimeric transcription activators
	(cf.~\S\ref{ss:HrpRS}).
\item
	RNA \AND-gates
	\cite[\href{https://www.nature.com/articles/nature23271/figures/2}{Fig.~2e}]{GreenETAL2017}
	instead of the protein heterodimer.
\item
	\href{https://bit.ly/wsgxfDv}{RNA-only construct}
	inspired by
	\cite[\href{https://www.nature.com/articles/nature23271/figures/2}{Fig.~2e}]{GreenETAL2017},
	where
	each input expresses its own mRNA,
	whose
	5' tail opens its RBS stemloop;
	%
	each input also expresses
	complementary RNAs
	that
	sequester other tails 
	thus closing up each other's RBS;
	%
	upon all three inputs,
	all complementary RNAs 
	assemble into a ring,
	allowing the 5' tails to reopen the RBS stemloop.
\item
	Three orthogonal resettable recombinases
	as in 
	\cite[\href{https://www.nature.com/articles/s41598-017-07386-3/figures/2}{Fig.~2g}]{ChiuJiang2017}.
\item
	\TODO{Describe the Cas variant}
\end{itemize}

	

\ra{other notes:}

\ra{full adder in ``mammalian'' cells \cite{WeinbergETAL2017} (\url{https://www.addgene.org/87552/}),
communicating \cite{AuslaenderETAL2017}}



%


\TODO{migrate what's useful to \S\ref{ss:HrpRS}:}

In \cite[\href{https://bmcbiol.biomedcentral.com/articles/10.1186/s12915-015-0146-0/figures/5}{Fig.~5}]{Wong2015}, 
a \XOR gate is built with two plasmids. 
%
On the first plasmid, 
a constitutive promoter \TODO{need to specify at some point}
promotes the synthesis of proteins AraC and RhaS.
%
% Ara
\ra{
AraC is an arabinose-induced transcription activator
of the promoter pBAD upstream of HrpS. 
%
% Rha
Similarly, RhaS activates the promoter pRHAB in the presence of inducer rhamnose and then synthesis of HrpR.
%
HrpR and HrpS form a heterodimer HrpRS
that binds to and activates the promoter pHrpL 
upstream of the repressor $\lambda$CI 
on the second plasmid.
%
At the same time, inducers arabinose and rhamnose 
each promote the expression
of the reporter RFP on the second plasmid via
the promoters pBAD and pRHAB,
which are, however,
modified to allow $\lambda$CI to bind
and prevent expression.
%
In this way, 
RFP is expressed when either one of them is present but not both. 

It is noted \TODO{by whom, where \hh{in this exact cited paper, when they discuss why the copy number is like this} 
\TODO{what are the copy numbers?} 
\ra{if you make a specific claim, provide a specific reference [delete when read]}
}
that the promoter pHrpL is relatively weak.
%
The authors propose to 
increase the copy number of 
the second plasmid (where the promoter is located)
in order to
increase the dynamic range of the activator dimer.
}


%\begin{figure}
%    \centering
%    \includegraphics{circuits/3_input_xor.png}
%    \caption{3 input \XOR logic gate as cascading \XOR gates}
%    \label{f:my_label}
%    \ra{cf. new figure \ref{f:logical-subtractor01}}
%\end{figure}

\hh{
The first 2-\XOR (A and B) use the above implementation 
(i.e.~with heterodimer HrpRS) to regulate the expression of pchB. 
%
The 3rd inducer C then triggers the expression of pchA.
%
Similarly, only when pchB and pchA are co-expressed, the dimer pchBA can trigger the synthesis of salicylic acid, which then regulates the activator NahR to activate the expression of repressor McbR \cite{Rey2005}.

}

\begin{figure}
    \centering
    \includegraphics[width = \linewidth]{gene_circuits_xor/P1.png}
    \includegraphics[width = \linewidth]{gene_circuits_xor/P2.png}
    \includegraphics[width = \linewidth]{gene_circuits_xor/P3.png}
    \includegraphics[width = \linewidth]{gene_circuits_xor/P4.png}
    \caption{Construct of layered \XOR gate. Inducer A, B and C are Ara, Rha and IPTG, which regulate the activity of pBAD, pRHAB and pTAC respectively. Here the tagged promoters refers to the ones that can be blocked by the upstream repressor. \hh{This is just a design sketch taht i came up with and drew with \url{https://sboldesigner.github.io}. For the final report we would use something looked nicer for the design diagram}
    \ra{if you keep them online please provide a url and/or upload all relevant files (not just the figures); also: vector graphics [delete when read]}
    }
    \label{f:gene_circuit_xor}
\end{figure}


\ra{Note: we don't do PNG b/c that doesn't scale well (neither up nor down). unfortunately, www.circuit-diagram.org doesn't export eps or pdf, but it does export svg, from which I make pdf [delete when read]}

%






\section{Simulations}

\subsection{PlayerA}

\url{https://github.com/numpde/ibiocomp/tree/main/code/20201231_SymBio_All/PlayerA}

\TODO{Some details missing:
\href{https://drive.google.com/file/d/1X1uEVNkUiXZqNz7C7m36YjW-g1XctIH0}{click for figure (ignore black part)}}

\subsection{Bit 0}

\url{https://github.com/numpde/ibiocomp/tree/main/code/20201231_SymBio_All/Bit0}

\TODO{this could have the same problem as described in Bit 1}

\subsection{Bit 1}

\url{https://github.com/numpde/ibiocomp/tree/main/code/20201231_SymBio_All/Bit1}

\subsubsection*{d1} \label{ss:d1}

Consider bit 1 of Subtractor A.
%
Suppose \ce{s_1 r_1 c_1 = 101},
in which case \ce{c_2 = 0}.
%
Suppose a transcription factor \ce{Y}
encoding \ce{c_2},
promotes expression of \ce{Xis} in the memory subunit.
%
%
When \ce{w_B} is up,
access to memory is on.
%
Since we want \ce{c_2 = 0},
the amount of \ce{Y} is low,
but
around the timepoint when 
\ce{w_B} is turned off
and the inputs begin to change,
\ce{Y} can experience a transient bump.
%
This is sufficient to produce \ce{Xis}
and thus to cause an unintended \emph{reset}
towards \ce{c_2 = 1}.
%
We found two ways to mitigate this
in the computational model.
%
One is to condition the expression of \ce{Xis}
on \ce{w_B} also
--
but this requires \ce{w_B} to go down
before other change
and could be sensitive to degradation rates.
%
A more robust way
is a buffer for \ce{Y}
(a ``shelf'')
that 
%
i)
has high affinity for \ce{Y};
%
ii)
sequesters 
a large amount of \ce{Y}
(about 30$\times$\href{https://en.wikipedia.org/wiki/EC50}{EC50} of $\ce{Y} \act \ce{Xis}$);
%
iii)
releases \ce{Y} for degradation when \ce{w_A} is on.
%
In this way, 
a transient increase in \ce{Y} is shelved
but a sustained signal 
(while \ce{w_B} is up)
will overflow the shelves and activate 
transcription of \ce{Xis}.
%
%
This is illustrated in Fig.~\ref{f:symbio-d1-tempo}.

\TODO{\ra{Issue: how to implement this ``Shelf''?}}
\ra{%
Note: This seems similar to
\cite[\href{{https://www.pnas.org/content/pnas/104/8/2643/F1.large.jpg?width=800&height=600&carousel=1}}{Fig.~1}]{WeberETAL2007},
also
\cite{NilgiriwalaETAL2014}.
}


\begin{figure}[phbt]
\centering

\begin{overpic}[width=0.99\textwidth]{Bit1/response_d1_tempo}
\put (50, 30) {%
	\scalebox{0.7}{%
		\begin{tabular}{c|ccc|c}
		    & \ce{\#s_1} & \ce{\#r_1} & \ce{\#c_1} & $\approx \ce{\#d_1}$
		    \\
		    \hline
		    \ce{B_1} & 10 & 10 & 10 & 0 
		    \\
		    \ce{A_1} & {\color{gray}$0.01$} & {\color{gray}$0.01$} & {\color{gray}$0.01$} & 10 
		    \\
		    \ce{B_2} & 10 & 10 & {\color{gray}$0.01$} & 0 
		    \\
		    \ce{A_2} & {\color{gray}$0.01$} & {\color{gray}$0.01$} & {\color{gray}$0.01$} & 0
		    \\
		\end{tabular}
	}
}
\end{overpic}

\caption{%
Subtractor A, bit 1, output \ce{\#d_1}
across phases \ce{B_1}-\ce{A_1}-\ce{B_2}-\ce{A_2}
(with interphases).
%
The table shows the inputs for each phase.
%
As required,
we have
$\#\ce{d_1} \approx 10$ 
during the recall phase \ce{A_1}
and
$\#\ce{d_1} \approx 0$ 
during the recall phase \ce{A_2}.
%
Observe the slight delay in the rise of Y
at the beginning of \ce{B_1} while it is being ``shelved''
and a sustained supply by ``unshelving'' at the end
that helps maintaining Xis;
conversely,
the leakage of Y is ``shelved''
during \ce{B_2}
preventing unwanted expression of Xis.
%
See also \S\ref{ss:d1}/p.\pageref{ss:d1}.
%
}
%
\label{f:symbio-d1-tempo}
\end{figure}

%

\begin{figure}[hpbt]
	\begin{tabular}{cc|cc}
		\multicolumn{2}{c|}{\ce{\#d_1}} & \multicolumn{2}{c}{\ce{\#c_2}}
		%
		\\
		%
		\ce{\#c_1} = 0.01 & \ce{\#c_1} = 10 &
		\ce{\#c_1} = 0.01 & \ce{\#c_1} = 10 
		%
		\\
		%
		\includegraphics[width=0.22\textwidth]{Bit1/response_d1_final__c1_in=0.1}
		&
		\includegraphics[width=0.22\textwidth]{Bit1/response_d1_final__c1_in=10}
		&
		\includegraphics[width=0.22\textwidth]{Bit1/response_c2__c1_in=0.1}
		&
		\includegraphics[width=0.22\textwidth]{Bit1/response_c2__c1_in=10}
	\end{tabular}
	%
	\caption{%
		Response surfaces of the Subtractor
		for varying \ce{\#r_1} (vertical) and \ce{\#s_1} (horizontal).
		%
		The output \ce{\#d_1}
		is measured after a compute-, inter- and readout-phase,
		cf.~Fig.~\ref{f:symbio-d1-tempo}.
		%
		The log-colorbar goes from $10^{-3}$-or-less (white) to $10^1$ (black).
	}
	%
	\label{f:sub_response}
	%
	\TODO{caption}
\end{figure}

%

\section{Discussion}

\TODO{risks}


\footnotesize
\bibliographystyle{apalike}
\bibliography{refs}
\normalsize

\clearpage



\section*{Appendix}

\subsection{Subtractor module}

Fig.~\ref{f:logical-subtractor01} 
shows the main principles
behind 
the Subtractor module (bit 0 and bit 1),
but our implementation
differs in the details.
%
For example, we use a reversible recombinase
for the memory module, see \S\ref{ss:memory}/p.\pageref{ss:memory}.

% https://www.circuit-diagram.org/editor/c/075f8cf9d084400b94f1384cc1c3ba96
\begin{figure}[phbt]
\centering
\includegraphics[width=0.9\linewidth]{circuits/Logical-HalfSubtractor0.svg.pdf}
\caption{%
Bit~0 and Bit~1
of
Subtractor~A.
%
The result of the compute unit is written to the memory unit
(e.g.~a bistable switch)
when the signal \ce{w_B} is present.
%
The value of the memory unit is
forwarded to the output
when the signal \ce{w_A} is present.
}
\label{f:logical-subtractor01}
\end{figure}




\subsection{Cross-inhibition \AND gate} \label{s:x-inh}

\TODO{explain}

\ra{
this is implemented in \ce{c_2}
[\href{https://i.ibb.co/86S2qfL/2021-01-17-01-42-06.png}{fig}]
}


\subsection{Kinetics}


\subsubsection*{AmtR}

We therefore assume
the mechanism
\begin{subequations}
\[
	\ce{
		2 \protein{AmtR} + \promoter{AmtR}
		& <=>>
		\protein{AmtR}_2 \with \promoter{AmtR}
	}
	%
	\\
	%
	\ce{
		\promoter{AmtR} 
		& ->
		\promoter{AmtR} + \protein{Output}
	}
\]
\end{subequations}
and
the kinetics:
\begin{subequations}
\[
	\label{e:AmtR_Act}
	%
	\promoter{AmtR} 
	& =
	\frac{1}{1 + k_{\ref{e:AmtR_Act}} \protein{AmtR}^{n_{\ref{e:AmtR_Act}}}}
	\promoter{AmtR}_\text{total}
	%
	% here, k = forward / backward
	%
	,
	\\
	%
	\label{e:AmtR_Out}
	%
	\tfrac{\d}{\d{t}}
	\protein{Output} 
	& =
	k_{\ref{e:AmtR_Out}}
	\promoter{AmtR}
	.
\]
\end{subequations}


\subsubsection*{3OC6}


We write
$\protein{LuxR^\star}$ for $\signal{3OC6\text{-}HSL} \with \protein{LuxR}$
and
$\promoter{Lux^\star}$ for $\protein{LuxR^\star} \with \promoter{Lux}$.
%
%\TODO{
%Let $\protein{LuxR^\star}$ denote 
%the activated form
%$\signal{3OC6}:\protein{LuxR}$.
%}
%
%
We assume the mechanism:
%
\begin{subequations}
\[
	\ce{
		\signal{{3OC6}} + \protein{LuxR}
		& <=>>
		\protein{LuxR^\star}
	}
	\\
	\ce{
		\protein{LuxR^\star} + \promoter{Lux}
		& <=>>
		\promoter{Lux^\star}
	}
	\\
	\ce{
		\promoter{Lux^\star}
		& ->
		\promoter{Lux^\star} + \protein{Output}
	}
	.
\]
\end{subequations}
%
%=======
%	\protein{LuxR^\star} + \protein{RNApol} + \promoter{P_{Lux}}
%	& \longleftrightarrow
%	\protein{LuxR^\star} + \protein{RNApol} : \promoter{P_{Lux}}
%	\\
%	\protein{RNApol} : \promoter{P_{Lux}}
%	& \longrightarrow
%	\protein{RNApol} + \promoter{P_{Lux}} + \protein{Output}
%>>>>>>> c8a0f57bdb0a7b80747263819cbb64562e087765
%\TODO{
%\[
%	\signal{3OC6} + \protein{LuxR}
%	& \longleftrightarrow
%	\protein{LuxR^\star}
%	\\
%	\protein{LuxR^\star} + \protein{RNApol} + \promoter{Lux}
%	& \longleftrightarrow
%	\protein{LuxR^\star} + \protein{RNApol} : \promoter{Lux}
%	\\
%	\protein{RNApol} : \promoter{Lux}
%	& \longrightarrow
%	\protein{RNApol} + \promoter{Lux} + \protein{Output}
%\]
%}
% https://chem.libretexts.org/Bookshelves/Biological_Chemistry/Supplemental_Modules_(Biological_Chemistry)/Enzymes/Enzymatic_Kinetics/Sigmoid_Kinetics
%
%
%
and the kinetics:
%
\begin{subequations}
\[
	\label{e:LuxR_Act}
	%
	\protein{LuxR^\star} 
	& =
	\frac{
		\signal{{3OC6}}
	}{
		k_{\ref{e:LuxR_Act}} + \signal{{3OC6}}
	}
	\protein{LuxR}_\text{total}
%	\TEXT{initially}
%	\protein{LuxR^\star} = 0
	,
	%
	\\
	%%
	\label{e:P_Lux_Act}
	%
	\promoter{Lux^\star} 
	& =
	\frac{
		\protein{LuxR^\star}
	}{
		k_{\ref{e:P_Lux_Act}} + \protein{LuxR^\star}
	}
	\promoter{Lux}_\text{total}
%	\TEXT{initially}
%	\promoter{Lux^\star} = 0
	,
	%
	\\
	%
	\label{e:P_Lux_Out}
	%
	\tfrac{\d}{\d{t}}
	\protein{Output}
	& =
	k_{\ref{e:P_Lux_Out}} \promoter{Lux^\star}
	.
\]
\end{subequations}
%
%
In combination, we have
\[
	\tfrac{\d}{\d{t}}
	\protein{Output}
	=
	k_{\ref{e:P_Lux_Out}}
	\promoter{Lux}_\text{total}
	\frac{
		\signal{} \protein{}_\text{total}
	}{
		k_{\ref{e:LuxR_Act}} k_{\ref{e:P_Lux_Act}}
		+
		k_{\ref{e:P_Lux_Act}} \signal{}
		+
		\signal{} \protein{}_\text{total}
	}
	.
\]


\subsubsection*{Sal}


We write
$
	\promoter{Sal^\star} :=
	\signal{Sal}_4 \with \protein{NahR}_4 \with \promoter{Sal}
$
for the active form.
%
%
We assume the mechanism
%
\begin{subequations}
\[
	\ce{
		\signal{Sal} + 
		\signal{Sal}_{k - 1} \with \protein{NahR}_4 \with \promoter{Sal}
		& <=>>
		\signal{Sal}_k \with \protein{NahR}_4 \with \promoter{Sal}
	}
	,
	\quad \text{k} = 1, 2, 3, 4,
	%
	%
	\\
	%
	%
	\ce{
		\promoter{Sal^\star}
		&
		->
		\promoter{Sal^\star} + \protein{Output}
	}
\]
\end{subequations}
%
%Mechanism (here \protein{NahR^\star} denotes the activated (tetramer) form):
%\[
%	\signal{Sal} + \protein{NahR}
%	& \longleftrightarrow
%	\protein{NahR^\star}
%	 \\
%	 \protein{NahR^\star} + \protein{RNApol} + \promoter{P_{Sal}}
%	& \longleftrightarrow
%	\protein{NahR^\star} + \protein{RNApol} : \promoter{P_{Sal}}
%	\\
%	\protein{RNApol} : \promoter{P_{Sal}}
%	& \longrightarrow
%	\protein{RNApol} + \promoter{P_{Sal}} + \protein{Output}
%	.
%\]
%
%
which we implement as:
%
\begin{subequations}
\[
	\label{e:NahR_Act}
	%
	\promoter{Sal^\star}
	& =
	\frac{
		\signal{}^{n_{\ref{e:NahR_Act}}}
	}{
		k_{\ref{e:NahR_Act}} + \signal{}^{n_{\ref{e:NahR_Act}}}
	}
	\promoter{Sal}_\text{total}
	,
	%
	%
	\\
	%
	%
	\label{e:NahR_Out}
	%
	\tfrac{\d}{\d{t}}
	\protein{Output}
	& =
	k_{\ref{e:NahR_Out}} 
	\promoter{Sal^\star}
	.
\]
\end{subequations}






\clearpage

\section*{Final TODOs}

\TODO{spacing of act and rep}

\TODO{check that Table \ref{t:signals} is correct}


\clearpage

\SHOWTODOS




\leavevmode\vfill{\tiny\color{lightgray}\hfill{\DTMnow}}
\end{document}





