\documentclass[12pt,notitlepage]{article}

% Overleaf project:
% https://www.overleaf.com/project/5fdb880ef955c3369964d601

% View-only link:
% https://www.overleaf.com/read/hkybwqtbtvjb

\usepackage{amsmath, amsfonts, amssymb, mathtools}
\usepackage[svgnames]{xcolor}
\usepackage{datetime2}
\usepackage[
	colorlinks=true, 
	citecolor={DarkRed}, urlcolor={DarkBlue}, linkcolor={DarkBlue},
]{hyperref}


\usepackage{natbib}

% https://tex.stackexchange.com/questions/3802/how-to-get-doi-links-in-bibliography
% \usepackage{doi}

% 
\usepackage[version=4]{mhchem}

\usepackage{fullpage}

% Paragraph spacing
\usepackage{parskip}
%\usepackage{enumitem}

\usepackage{xspace}

\usepackage{graphicx}
\graphicspath{{../code/20201231_SymBio_All/}{../images/}}
\DeclareGraphicsExtensions{.pdf,.eps,.png}

% https://tex.stackexchange.com/questions/202046/width-of-the-caption-of-a-figure
% https://tex.stackexchange.com/questions/29039/how-to-limit-the-figure-caption-width
\usepackage[margin=10px]{caption}

% https://tex.stackexchange.com/questions/20792/how-to-superimpose-latex-on-a-picture
\usepackage[percent]{overpic}

%\usepackage{epstopdf}

% Tables (the order matters here)
\usepackage{makecell}
\usepackage{booktabs}
\usepackage{arydshln}

% https://tex.stackexchange.com/questions/109467/footnote-in-tabular-environment
\usepackage{footnote}
\makesavenoteenv{tabular}
\makesavenoteenv{table}

% https://tex.stackexchange.com/questions/10130/use-the-values-of-title-author-and-date-on-a-custom-title-page
\usepackage{authoraftertitle}

% https://en.wikibooks.org/wiki/LaTeX/Footnotes_and_Margin_Notes#Margin_Notes
\usepackage{marginnote}

% For editing purposes:
%\usepackage[margin=10pt]{geometry}

% https://latex.org/forum/viewtopic.php?t=10456
\usepackage{titlesec}
\titleformat{\subsubsection}[runin]% runin puts it in the same paragraph
{\normalfont\bfseries}% formatting commands to apply to the whole heading
{\thesubsubsection}% the label and number
{}% space between label/number and subsection title
{}% formatting commands applied just to subsection title
[.]% punctuation or other commands following subsection title


\newcommand{\TODO}[1]{\textrm{\color{red}TODO: #1}}
\input{todomech}


\renewcommand{\d}{\mathrm{d}}

\newcommand{\NOT}{\ensuremath{\mathop{\mathsf{not}}}\xspace}
\newcommand{\AND}{\ensuremath{\mathop{\mathsf{and}}}\xspace}
\newcommand{\OR}{\ensuremath{\mathop{\mathsf{or}}}\xspace}
\newcommand{\XOR}{\ensuremath{\mathop{\mathsf{xor}}}\xspace}

\newcommand{\TEXT}[1]{\quad\text{#1}\quad}
\newcommand{\with}{\text{$\,{:}\,$}}

\newcommand{\cbra}[1]{{\ensuremath{\color{gray}{#1}}}}
\newcommand{\signal}[1]{{{\cbra{\langle}\ce{#1}\cbra{\rangle}}}}
\newcommand{\protein}[1]{{{\cbra{(}\ce{#1}\cbra{)}}}}
\newcommand{\promoter}[1]{{{\cbra{[}\ce{#1}\cbra{]}}}}

% https://tex.stackexchange.com/questions/543953/arrow-with-blunted-end-head-in-math-mode
\newcommand{\act}{\ {\ensuremath{\mathbin{\to}}}\ }
\newcommand{\rep}{\ {\ensuremath{\mathrel{\raisebox{-.3ex}{\rotatebox{90}{\scalebox{1}[1.2]{$\bot$}}}}}}\ }

\def\[#1\]{\begin{align}#1\end{align}}

% https://tex.stackexchange.com/questions/114113/how-to-label-text-with-equation-number
\newcommand{\eqnum}{\leavevmode\hfill\refstepcounter{equation}\textup{{(\theequation)}}}

\newcommand{\starlink}[1]{\textsuperscript{\makebox[0pt]{\href{#1}{\color{white}$\star$}}}}

\newcommand{\hh}[1]{{\color{Purple}#1}}
\newcommand{\ra}[1]{{\color{Blue}#1}}


\title{ibiocomp project}
\author{RA \& HH}
\date{\today}
\newcommand{\linktodoc}{http://bit.ly/15sticks}



\begin{document}

\maketitle

% \ra{LOCKED 'COS WORKING WITH ANOTHER COPY (WRITE EMAIL IF NEEDED)}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Introduction}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\subsection{The game of 15 sticks}

We design genetic circuits
that play the following game:
%
\begin{quote}
	Two players start with 15 sticks
	and alternatingly 
	take 1, 2 or 3 sticks.
	\\
	The last player to take a stick loses.
\end{quote}

Because a player can complement
the number of sticks taken by the opponent,
there is a deterministic winning strategy
of the first-mover
as shown in Table \ref{t:logical-playera}.
%
%
We implement this strategy for both players.


\subsection{General design} \label{s:general}

We encode in binary 
the number of sticks left as
\[
    \label{e:s0123}
    %
    \text{number of sticks left} = \ce{8 s_3 + 4 s_2 + 2 s_1 + s_0}
    .
\]
%
%
By the 4-periodicity of the winning strategy,
only the two lowest bits \ce{s_1}/\ce{s_0}
matter to the player.
%
%
The response of a player is a number 1, 2, or 3,
which we encode in binary as
\[
    \label{e:r01}
    %
    \text{number of sticks taken by a player} = \ce{2 r_1 + r_0}
    .
\]
%
The bits in \eqref{e:s0123}--\eqref{e:r01}
are transmitted 
between compute units
by 
intercellular signaling molecules,
cf.~\S\ref{ss:experiment}.
%
Although our players are essentially identical,
we model them separately and with one difference:
Player A (B) becomes active when 
the extracellular signal \ce{w_A} (\ce{w_B}) is supplied.

%

The \emph{subtractor} module
keeps track of the number of sticks left \eqref{e:s0123}.
To appreciate the key design issue, suppose
we manually transmit the response \ce{r_1 r_0 = 01}
of a player
into the medium of the subtractor.
It should compute the new number of the sticks,
yet, it has to do so once
and not keep subtracting the value \ce{r_1 r_0}
that is still present.
One solution is single-use subtractors
but then the result of each computation
has to be imparted to an unused subtractor.
Barring excessive manual intervention,
a feasible design
seems to require some separation of
reading/storing the current state,
computing the new state and communicating it
--
temporally or spatially.
%
%
%
We propose 
a subtractor composed of two parts A and B,
and
a workflow 
%of Phase A, Interphase, Phase B, Interphase, etc.,
as shown in Table \ref{t:workflow}.
%
%
%
Thus,
in Phase A characterized by the supply of 
the master signal \ce{w_A},
Subtractor A
emits the current state \ce{s},
Player A responds with \ce{r},
while
Subtractor B
reads \ce{s}/\ce{r}
and
computes the new state % \ce{s} 
{silently}
and remembers the result
in order to emit it in the subsequent Phase B.
%
%
We use the variables \ce{d_3}/\ce{d_2}/\ce{d_1}/\ce{d_0}
for this internal state.
%
%
The medium is cleared, before
Phase B is initiated by
supplying the master signal \ce{w_B}.
%
%
In short,
we require
a gated memory module and a delayed response,
cf.~\S\ref{ss:sub}.
%
%

%

%%% TABLE %%%
\input{t_workflow}



\subsection{Conventions}



%
As shown in Table \ref{t:signals},
we encode \ce{w_A} by L-arabinose,
\ce{w_B} by IPTG, etc.
%
Our convention is \ce{w_A = 1}
if 
the corresponding messenger molecule L-arabinose 
is present in the medium or in the cell 
in sufficiently high quantity,
and \ce{w_A = 0} otherwise.
%
We write \ce{\#w_A} for
the \emph{number} of L-arabinose molecules 
in the numerical simulations.
%
Where suitable,
we indicate the semantics of an element thus:
\signal{signal},
\protein{protein} -- often a transcription factor,
and
\promoter{promoter};
%
this is consistent with 
the symbols that we employ in
the SimBiology schemes.
%
By \signal{w_A} we mean 
``the signal molecule that encodes \ce{w_A}'',
i.e.~L-arabinose.
%
Ditto for the other signals.

%
%

Abbreviations:
RBS = ribosome binding site,
QS = quorum sensing,
AHL = acyl homoserine lactone,
txn = transcription,
tln = translation,
{3OC6} = {3OC6}{-}HSL (etc.),
SM = supplementary material,
DNF = disjunctive normal form.



\subsection{Logic circuits}


\subsubsection*{The players} \label{ss:players}

Both players
follow the same deterministic strategy,
which is a winning strategy for the first-mover.
%
They are identical except
that player A responds when \ce{w_A} is present
and player B responds upon \ce{w_B}.
%
The logic is shown in Table \ref{t:logical-playera}.


\subsubsection*{Subtractor} \label{ss:sub}


In Phase A,
Subtractor B 
senses the current number of sticks \eqref{e:s0123},
emitted by Subtractor A,
and 
the response \eqref{e:r01} from Player A,
and
computes the difference \ce{d}.
%
It has four compute units Bit~0 to Bit~3
that are also
spatially separated (cf.~\S\ref{ss:experiment}). 
%
The subtraction proceeds from Bit~0 to Bit~3
(cf.~Table~\ref{t:substeps}).
%
Bit~0 senses the current \ce{s_0}/\ce{r_0}
and computes \ce{d_0} (silently)
and the carry flag \ce{c_1} (emitted immediately).
%
The carry flag is passed on to Bit~1.
%
Bit~1 subtracts \ce{r_1} and \ce{c_1} from \ce{s_1}. 
%
The resulting carry flag \ce{c_2} 
is passed on to Bit~2, 
which subtracts \ce{c_2} from \ce{s_2}.
%
Then Bit~3 subtracts the resulting carry flag
\ce{c_3} from \ce{s_3}.
%
Each of
Bit 0/2/3,
is in fact a ``half-subtractor'' with 2 inputs,
whilst
Bit~1 is a ``full subtractor'' with 3 inputs.
%
%
%
See
Table~\ref{t:subtractor1}
for their truth table.
%
Fig.~\ref{f:logical-subtractor01}
shows a reference logic circuit 
for Bit~0 and Bit~1
that illustrates the principle of delayed response.





%%% TABLE %%%
\input{t_logical-playera}


%%% TABLE %%%
\input{t_subtractor1}








\subsection{Experimental design} \label{ss:experiment}

We distinguish
\emph{extracellular} chemical signals 
that are supplied by the experimenter
(the master signals \ce{w_A}/\ce{w_B})
and
\emph{intercellular} signals
that are produced and sensed by cells.
%
%
%
For an experiment in a single test tube 
we need
orthogonal intercellular channels:
\emph{three} for \eqref{e:s0123}
($s_3$ has no downstream),
\emph{two} for \eqref{e:r01},
and
\emph{three} for the carry flags
(see \S\ref{ss:sub}).
%
One channel could be spared 
at the expense
of a more involved computation of 
the last useful carry flag~$c_3$.
%
Thus we need \emph{eight} orthogonal intercellular channels,
cf.~Table \ref{t:substeps}.
%
In \cite{DuETAL2020},
ten de novo intercellular signaling pathways
were designed
and
four known QS pathways
were optimized,
however,
only \emph{eight} worked well orthogonally
on the sensor level 
and 
seven on the promoter level
\cite[\href{https://www.nature.com/articles/s41467-020-17993-w/figures/3}{Fig.~3c/g}]{DuETAL2020}.
%
We have assigned the signals
to minimize crosstalk,
in particular 
avoiding LuxR and RpaR in one cell
as warned in \cite[p.6]{DuETAL2020}.
%
The signals are listed in
Table~\ref{t:signals}.
%
%
%

%%% TABLE %%%
\input{t_signals.tex}


%

\ra{
To facilitate debugging
we partition the workflow
with spatial compartments
that require manual transmission of 
the medium.
%
%
Unfortunately, this does not automatically reduce
the number of intercellular channels needed.
}


\TODO{what species are we working with}


\TODO{experimental protocol:}
%
In phase A.
%
Add \ce{w_A} everywhere.
%
Then:
%
\begin{enumerate}
\item 
    Admix the medium of Bit~0 and Bit~1 
    (containing the signals \ce{s_0} and \ce{s_1})
    to Player~A,
    wait for Player~A to respond
    (with \ce{r_0}/\ce{r_1}).
    
\item
    Admix the medium of Player~A
    (with \ce{r_0}/\ce{r_1}/\ce{s_0}/\ce{s_1})
    to Bit~0, wait for response
    (with \ce{c_1}).
    
\item
    Admix the media of Player~A and Bit~0
    (with \ce{r_0}/\ce{r_1}/\ce{s_0}/\ce{s_1}/\ce{c_1})
    to Bit~1, wait for response
    (with \ce{c_2}). 
    
\item
    
\item
    Admix the medium from Bit~3 to Bit~4.
    
\end{enumerate}




\begin{table}[hpbt]
    \centering
    %
    % Table made with
    % https://deepnote.com/project/998c51eb-bb3f-4a0f-88e3-c8b50d4d678a#%2Ft_phases.ipynb
    %
    \begin{tabular}{l||cc|ccc|ccc|ccc|cc}
    {} &      \ce{r_0} &      \ce{r_1} &    \ce{s_0} &      \ce{d_0} &      \ce{c_1} &      \ce{s_1} & \ce{d_1} &      \ce{c_2} &    \ce{s_2} &      \ce{d_2} &      \ce{c_3} &    \ce{s_3} &      \ce{d_3} \\ 
    \hline
    1 &  $\downarrow$ &  $\downarrow$ &  $\uparrow$ &               &               &    $\uparrow$ &          &               &             &               &               &             &               \\ 
    \hline
    2 &    $\uparrow$ &               &  $\uparrow$ &  $\downarrow$ &  $\downarrow$ &               &          &               &             &               &               &             &               \\ 
    \hline
    3 &               &    $\uparrow$ &             &               &    $\uparrow$ &  $\downarrow$ &          &  $\downarrow$ &             &               &               &             &               \\ 
    \hline
    4 &               &               &             &               &               &               &          &    $\uparrow$ &  $\uparrow$ &  $\downarrow$ &  $\downarrow$ &             &               \\ 
    \hline
    5 &               &               &             &               &               &               &          &               &             &               &    $\uparrow$ &  $\uparrow$ &  $\downarrow$ \\ 
    \hline
    \end{tabular}
    %
    %
    \caption{%
        Substeps of each phase A/B.
        Arrows: emit ($\uparrow$) and compute ($\downarrow$).
    }
    \label{t:substeps}
\end{table}


\TODO{mention a microfluids setup?}


We initialize the experiment 
in Phases A
by supplying
the master signal \signal{w_A}
along with
\signal{s_3}/\signal{s_2}/\signal{s_1}/\signal{s_0}
to Subtractor B
and
to Player A.

%

Refinements could include
a)
dealing with
a player who attempts to take more sticks than \eqref{e:s0123};
b)
dedicated sensor cells 
that report on signaling molecules 
in order to 
monitor the state of the game;
c)
a set of cells that emit 
\signal{s_3}/\signal{s_2}/\signal{s_1}/\signal{s_0}
upon one signal, say 
tetracycline%
\starlink{https://en.wikipedia.org/wiki/Tetracycline}.





\subsection{``Cite as'' and codes}

\MyTitle, \MyAuthor, \MyDate.
\href{\linktodoc}{{\small\texttt{\linktodoc}}}

See the link 
for the Matlab/SymBiology files
and 
for the PDF file with sharp figures.


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Genetic toolbox} \label{s:genetic}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\subsection{Constitutive parts} \label{ss:const}

\subsubsection*{Constitutive promoter} \label{ss:Pc}

\ra{
The iGEM part 
\href{http://parts.igem.org/Part:BBa_J23100}{BBa\_J23100}
is a strong constitutive \emph{E.~coli} $\sigma^{70}$ promoter
\cite{iGem_Pc}.
%
We refer to it as \ce{P_c}.
%
\TODO{check that citation is correct}
}

\hh{
Constitutive promoter activity requires \ra{a} RNA polymerase holoenzyme
but no transcription factors. 
%
Since we work with \emph{E.~coli},
the constitutive promoters are all from \emph{E.~coli}, 
which utilize the bacteria's internal RNA polymerase for gene expression. 

}

%\url{http://parts.igem.org/Promoters/Catalog/Constitutive}
%\cite{ShimadaETAL2014}


\subsubsection*{cI} \label{ss:cI}

\ra{
The optimized and orthogonal variant
$\ce{cI^*} := \ce{cI_{5G6G,P}}$
of the transcription factor 
\href{https://www.uniprot.org/uniprot/P03034}{$\lambda$~\ce{cI}}
from
\cite[\href{https://www.nature.com/articles/ncomms13858/figures/4}{Fig.~4c/d}]{BroedelJaramilloIsalan2016}
requires no cofactors
and
shows 
about 10-fold activation of the promoter
$\promoter{cI^*_+} := \promoter{M,{5G6G}}$
and 
a similar-fold repression of 
$\promoter{cI^*_-} := \promoter{{5G6G}}$.
%
%
%To improve the dynamic range of 
%$\promoter{cI^*_+} \act \text{mRNA}_1$
%we propose 
%that
%$\promoter{cI^*_-}$ express
%an inhibitory RNA that is complementary
%to the RBS region of $\text{mRNA}_1$.
}




\subsection{3-input gates by Cello} \label{ss:cello}

In
\cite{NielsenETAL2016}
we found these gates
among good performers:
%
%\begin{itemize}
%\item
    $
    	\ce{r_0} = 
    	\mathop{\mathsf{%
    	\href{https://science.sciencemag.org/content/sci/352/6281/aac7341/F5/graphic-6.large.jpg}{0x0E}%
    	}}
    	(\ce{s_0}, \ce{s_1}, \ce{w_A})
    $;
%\item
    $
    	\ce{r_1} =
    	\NOT
    	\mathop{\mathsf{%
    	\href{https://science.sciencemag.org/content/sci/352/6281/aac7341/F5/graphic-5.large.jpg}{0xF6}%
    	}}
    	(\ce{s_0}, \ce{s_1}, \ce{w_A})
    $;
%item
    $
    	\ce{c_2}
    	=
    	\NOT
    	\mathop{\mathsf{%
    	\href{https://science.sciencemag.org/content/sci/352/6281/aac7341/F5/graphic-8.large.jpg}{0x8E}%
    	}}
    	(\ce{c_1}, \ce{r_1}, \ce{s_1})
    	=
       	\mathop{\mathsf{%
    	\href{https://science.sciencemag.org/content/sci/352/6281/aac7341/F5/graphic-5.large.jpg}{0x4D}%
    	}}
       	(\ce{c_1}, \ce{s_1}, \ce{r_1}) 
    $,
    which is
    symmetric in \ce{r_1}/\ce{c_1};
%end{itemize}
%
but not the 3-\XOR, see \S\ref{ss:3xor}.

%

We used their $\mathsf{0x0E}$ for \ce{r_0},
replacing the sensors 
and
replacing the repressor {PhlF}
(which clashes with the \ce{s_0} sensor) 
by {HlyIIR}
(also from \cite{NielsenETAL2016}).
%
The last operation
is a ``poor man's \OR gate'';
we let it produce \ce{cI^*}
(\S\ref{ss:cI})
that
triggers the synthesis of \signal{r_0}.

%

We observed that \ce{r_1} can reuse 
most of this circuitry
and implemented 
the remainder using ribocomputing
\TODO{explain}.

%

For \ce{c_2}, we took
$\NOT \mathsf{0x8E}$,
replacing the sensors
and 
replacing the final 
``$\NOT (\NOT (\cdot) \OR \NOT (\cdot))$''
by
a ribo-\AND gate with cross-inhibition
(see \S\ref{ss:xinh}/p.\pageref{ss:xinh}).



\subsection{Intracellular pathways (transducers)}


\subsubsection*{AmtR}

The TetR-like AmtR 
regulates
the nitrogen starvation response
in 
the Gram-positive 
\emph{C.~glutamicum}%
\starlink{https://en.wikipedia.org/wiki/Corynebacterium_glutamicum}
\cite{JakobyETAL2000}.
%
This repressor is released
by
the trimeric adenylylated 
% $\mathrm{P_{II}}$-type 
signaling protein GlnK
\cite{BeckersETAL2005, SevvanaETAL2017},
which is present upon
nitrogen starvation. 
%
The AmtR box and the regulon is 
characterized in 
\cite{BeckersETAL2005},
and
several variants are compared 
in \cite{MuhlETAL2009}.
%
%
%
It seems clear that 
AmtR sits on the DNA as a dimer
\cite{SevvanaETAL2017},
\cite[\S3.4.2]{Schwab2019}.

% Further literature
%\cite{MuhlETAL2009}
%\cite{BuchingerETAL2009}

\subsubsection*{...}

\TODO{continue}


\subsection{Extracellular signaling} \label{ss:wAB}

The two master signals \ce{w_A}/\ce{w_B}
are encoded by L-arabinose/IPTG,
and are supplied by the experimenter.
%
%
%
Following \cite[SM, VII.M]{NielsenETAL2016}/\cite{LeeETAL2007}
we posit
a truncated \protein{AraC^*}
that has reduced crosstalk with \signal{IPTG}.
%
%
%
\ra{
The genetic sequences can be found 
in
\cite[SM, Table~S8]{NielsenETAL2016}
for \protein{Ara^*} and \protein{LacI},
and
in
\cite[SM, Table~S9]{NielsenETAL2016}
for \promoter{BAD} and \promoter{Tac}.
}

% Ref on Ara>AraC>BAD: \cite{LeeETAL2007}, \cite{Schleif2000}
% Ref on IPTG>LacI>Tac: \cite{DykxhoornStpierreLinn1996}



\subsection{Intercellular pathways}

\ra{
Here we collect 
key aspects of 
the intercellular receivers.
%
%
3OC6, IV and 3OC12
belong to the AHL QS family,
whereas pC 
has a p-coumaric acid group 
rather than a fatty acyl group.
}
%
%


According to 
\cite[p.2598]{HirakawaETAL2011} \cite{Fuqua1996, Kylilis2018}
LuxR homologs are homodimeric.
%
\ra{now, which ones are LuxR homologs}
\hh{
luxR \& lasR \cite[Table~1]{Fuqua1996},  
rpaR \cite[p.2598]{HirakawaETAL2011}, 
bjaR \cite{Lindemann2011}. 
(Most -HSL receiver proteins are dimer If the signals have similar structure then so should the protein)
}

\ra{do all the receivers here dimerize and accept two signal molecules as in \cite[Fig.~1]{DuETAL2020}}

%
%
%
\ra{
Of note,
\cite[SM:p.8]{DuETAL2020} 
reports that sender cells
enter the log-phase with delay
compared to control samples:
up to 12h
for Sal and DAPG;
about 3h 
for 3OC6, 3OC12, pC and MMF;
no delay 
for IV and NG.
}
%
%
\ra{
For sequences, biosynthesis and induction curves 
we refer to
\cite[SM]{DuETAL2020}.
}


\subsubsection*{3OC6} \label{ss:3OC6}
 
The transcription activator LuxR
occurs in Gram-negatives
such as 
\emph{V.~fischeri}%
\starlink{https://en.wikipedia.org/wiki/Aliivibrio_fischeri}%
,
that is permeable to the (auto)inducer
3‐oxo‐C6‐homoserine lactone \signal{{3OC6}{-}{HSL}}.
%
%
It binds to the N-terminal of \protein{LuxR},
which otherwise inhibits its
functional C-terminal \cite{StevensDolanGreenberg1994}.
%
%
The purified C-terminal binds 
upstream of the \emph{lux} box 
(which is centered at $-42.5$bp \cite{EglandGreenberg1999});
however, 
together with the RNA Pol,
it protects the \emph{lux} box and the \emph{lux} operon
promoter
\cite{StevensDolanGreenberg1994}.


\subsubsection*{IV} \label{ss:IV}

\hh{
Isovaleryl-HSL (IV-HSL) is a branched-chain fatty acyl-HSL from soybean symbiont
\emph{B.~japonicum}%
\starlink{https://en.wikipedia.org/wiki/Bradyrhizobium_japonicum}
that regulate \protein{BjaR}.
In the presence of IV, dimerized BjaR activate \promoter{Bja}.
}

\subsubsection*{DAPG} \label{ss:DAPG}

\TODO{describe}

\subsubsection*{Sal} \label{ss:Sal}

According to 
\cite{SchellWender1986, HuangSchell1991},
the transcription activator \protein{NahR}
binds to the recognition site of \promoter{Sal} at
$-83$ to $-45$
without the inducer \cite[p.10837]{HuangSchell1991}.
%
%
In \cite{SchellBrownRaju1990},
it was suggested 
that the active configuration of \protein{NahR} is a tetramer,
while \cite{ParkLimShin2005}
reported that 
there could be three different complexes
$\protein{NahR} \with \promoter{Sal}$.
%
%
Possibly \cite{Peking2013},
$4 \times \protein{NahR}$ bind to the DNA,
and
transcription starts
once a $\signal{Sal}$
binds to each \protein{NahR}.
%
%Thus the inactive form of the promoter
%is $\protein{NahR}_4 \with \promoter{Sal}$.
%
%
\TODO{note: need \ce{P_c} for NahR} 

\TODO{
\ra{
From earlier:
\hh{NahR to activate the expression of repressor McbR \cite{Rey2005}}
Is \promoter{Sal} actually McbR ?
}
} \hh{No. McbR is a repressor protein. I think what I meant earlier is that NahR(TF) can activate Psal (promoter), which then activate the McbR gene (that encodes McbR protein). But it seems we do not need McbR now?}
\ra{it's in \cite{NielsenETAL2016} - but I don't think I used it}


\subsubsection*{pC} \label{ss:pC}

\ra{
P-coumaroyl-HSL/RpaR
works with p-coumaric acid rather than 
fatty acids of AHL/LuxR.
\TODO{$\leftarrow$ this is said above already, can remove?}
%
%Coming from anoxygenic phototrophic soil bacteria, 
%the luxIR-type pair, rpaI and rpaR 
It shows good orthogonality 
to many QS signals \cite{SchaeferETAL2008}.
}
%
\ra{
The \emph{R.~palustris}%
\starlink{https://en.wikipedia.org/wiki/Rhodopseudomonas_palustris}
%
transcriptional regulator RpaR,
when purified,
binds an inverted repeat element 
centered at $-48.5$bp
of its promoter
\cite{HirakawaETAL2011}.
%
%
%1
%There is
%pC-HSL-RpaR-activated antisense transcription of \protein{rpaR}.
%
%
The complex 
$\signal{pC} : \protein{RpaR}$
bound to the promoter
activates transcription
\cite[Discussion]{HirakawaETAL2011}.
}


\subsubsection*{MMF} \label{ss:MMF}

\TODO{describe}

\cite{ORourkeETAL2009}


\subsubsection*{3OC12} \label{ss:3OC12}


Two
3-oxo-dodecanoyl-homoserine lactone (3OC12-HSL)
molecules
bind ir\-rever\-sibly to the dimer \protein{LasR}
from \emph{P.~aeruginosa}%
\starlink{https://en.wikipedia.org/wiki/Pseudomonas_aeruginosa}%
. 
%
%
The activated LasR activates several promoters 
with high affinity 
with no or positive cooperativity % (Hill coefficients $\approx$ 1) 
\cite{SchusterUrbanowskiGreenberg2004, Grant2016}. 
%
%\cite[\href{https://www.embopress.org/doi/full/10.15252/msb.20156590}{p.1}]{Grant2016}
%
%
A custom promoter \promoter{Las^*} is used
in \cite[p.2]{DuETAL2020}.



\subsubsection*{NG} \label{ss:NG}

Naringenin (NG) is the main flavone from grapefruits
\cite{SiedlerETAL2014, iGEM14_TU_Darmstadt}.
%
It binds the homodimeric protein FdeR 
from \emph{H.~seropedicae}%
\starlink{https://en.wikipedia.org/wiki/Herbaspirillum_seropedicae}%
. 
%
There are two slots for NG,
whose binding
induces a conformational change of the dimer. 
%
The activated FdeR promotes 
the ``Flavanone degradation'' operon \promoter{fdeA}.



\subsection{Memory module} \label{ss:memory}

As explained in \S\ref{s:general}
(cf.~Table~\ref{t:workflow}/\ref{t:subtractor1} and Fig.~\ref{f:logical-subtractor01}),
we need a memory module 
to record and recall the result of the subtraction.
%
To that end we use a ``reversible'' recombinase.
%
%
Specifically,
the Bxb1 gp35/gp47
serine-integrase/excisionase
(Int/Xis)
direct the infection cycle of Bxb1 
in
\emph{M.~smegmatis}%
\starlink{https://en.wikipedia.org/wiki/Mycobacterium_smegmatis}%
.
%
Stoichiometry, start codons,
degradation rates (proteolysis tags)
and DNA substrate copy number 
were optimized 
in \emph{E.~coli}
in \cite{BonnetSubsoontornEndy2012}
to 
assemble a memory bit
that is
stable and responsive over several generations.
%
A promoter of choice is flanked 
by Int recognition sites
and
its direction is 
\emph{set}
by Int
and 
\emph{reset}
by Int+Xis.
%
Switching times of $\sim$4h were demonstrated
but the authors expected 30min to be realistic.
%
Their final construct
\cite[Fig.~4A]{BonnetSubsoontornEndy2012}
consists of an Int gene 
and
a Xis+Int operon
induced by two separate signals,
which facilitates stoichiometry optimization.
%
However, 
in our design,
an external pulse (say, \signal{w_B}) controls Int,
and
an internal compute unit (say, \ce{s_0 \XOR r_0}) controls Xis.
%
The \emph{reset} state corresponds to 
positive output, i.e.~the memory bit is \emph{on}.
%
%
The flanked promoter is activated by another
external signal (say, \signal{w_A}).


\subsection{Enhanced AND gate}

\subsubsection*{HrpR/HrpS} \label{ss:HrpRS}

\cite[\href{https://www.nature.com/articles/ncomms1516/figures/1}{Fig.~1}]{WangKitneyJolyBuck2011}
proposed 
an \AND gate in \emph{E.~coli}
using the heterodimeric transcription factor 
$\ce{HrpR} \with \ce{HrpS}$
from 
\emph{P.~syringae}%
\starlink{https://en.wikipedia.org/wiki/Pseudomonas_syringae}%
.
%
%The authors also estimated the parameters in 
%the Hill transfer function.
%
The dimer 
binds upstream of the promoter \ce{hrpL} 
and opens the $\ce{\sigma^{54}{-}RNAP{-}hrpL}$ transcription complex.
%
%
%
\ra{
The promoter is weak and saturates,
so increasing its copy number
improves the dynamic range
\cite[p.4]{Wong2015}.
}


\subsubsection*{Ribocomputing}

\cite[\href{https://www.nature.com/articles/nature23271/figures/2}{Fig.~2e}]{GreenETAL2017}
combined
two short input/sensor RNAs
\ce{A} and \ce{B}
into a ``trigger'' RNA
which opens an RBS-occluding stemloop
on the output mRNA.
%
%
%Let inputs \ce{A} and \ce{B} produce
%output \ce{mRNA_A} and \ce{mRNA_B}, resp.,
%whose RBS is auto-occluded by their own 5' tail.
%
%Let \ce{A} also produce a short RNA \ce{R^*_B} such that
%the 5' tail of \ce{mRNA_B} preferentially binds \ce{R^*_B},
%and vice versa.

\subsubsection*{Cross-inhibition} \label{ss:xinh}

Even in our idealized simulations
such \AND gates produce undesired leakage
within larger circuits.
%
%
%
We propose that
\emph{low} input \ce{a} 
{inhibit} the downstream of \ce{b},
and vice versa:
%
we let $\ce{a} \act \ce{cI^a}$
and
$\ce{b} \act \ce{cI^b}$
produce
two orthogonal 
bidirectional
transcription factors as in \S\ref{ss:cI}.
%
They, in turn, 
activate \promoter{cI^a_+}/\promoter{cI^b_+}
and 
repress \promoter{cI^a_-}/\promoter{cI^b_-}:
%
%
\begin{align}
    \begin{rcases*}
        \ce{B^*} \reflectbox{\act} 
        \promoter{cI^a_-} \reflectbox{\rep} \protein{cI^a} \act \promoter{cI^a_+} 
        \act \ce{A}
        \\
        \ce{A^*} \reflectbox{\act} 
        \promoter{cI^b_-} \reflectbox{\rep} \protein{cI^b} \act \promoter{cI^b_+} 
        \act \ce{B}
    \end{rcases*}
    %
    \act 
    \stackrel{\hspace{-10pt}\smash{\text{trigger RNA}}\hspace{-10pt}}{\ce{AB}}
    \act 
    \begin{array}{c}
        \text{opens RBS on} \\ \text{the output RNA.}
    \end{array}
\end{align}
%
%
Herein,
\ce{A} and \ce{B} 
combine to a trigger RNA \ce{AB} as in
\cite[\href{https://www.nature.com/articles/nature23271/figures/2}{Fig.~2e}]{GreenETAL2017},
and also have toeholds
for preferential binding to 
the complementary RNAs \ce{A^*} and \ce{B^*}.
%
%
The cross-inhibition is more effective
if
$\promoter{cI^a_-} > \promoter{cI^b_+}$
and
$\promoter{cI^b_-} > \promoter{cI^a_+}$
in terms of the output of the promoters.
%
%
See Fig.~\ref{f:sub_response}.


% see \S\ref{s:x-inh}.



% \subsection{Ribocomputing}
%
% \TODO{ribo section, could be unnecessary}


\subsection{3-input \texorpdfstring{\XOR}{XOR}} \label{ss:3xor}

The subtractor result bit
(cf.~Table~\ref{t:subtractor1})
\begin{align} \label{e:3xor}
	\ce{d_1} 
	= 
	\mathop{\mathsf{0x69}}
	(\ce{s_1}, \ce{r_1}, \ce{c_1})
	=
	\NOT
	\mathop{\mathsf{0x96}}
	(\ce{s_1}, \ce{r_1}, \ce{c_1})
	=
	\text{3-\XOR}
	(\ce{s_1}, \ce{r_1}, \ce{c_1})
	=
	((\ce{s_1} + \ce{r_1} + \ce{c_1}) \mathbin{\mathrm{mod}} 2)
\end{align}
is particularly difficult
because the result flips 
whenever any input is flipped.
%
We could not find it explicitly in \cite{NielsenETAL2016}.
%
%
It is a concatenation
of two \XOR gates (cf.~Fig.~\ref{f:logical-subtractor01})
with the DNF%
\starlink{https://en.wikipedia.org/wiki/Disjunctive_normal_form}:
$
	(\ce{s_1} \wedge \ce{r_1} \wedge \ce{c_1})
	\vee
	(s_1 \wedge \ce{\overline{r_1}} \wedge \ce{\overline{c_1}})
	\vee
	(\ce{\overline{s_1}} \wedge \ce{r_1} \wedge \ce{\overline{c_1}})
	\vee
	(\ce{\overline{s_1}} \wedge \ce{\overline{r_1}} \wedge \ce{c_1})
$.
%
We considered:
%
\begin{itemize}
\item
	Cascading 
	two \XOR gates
	built on tandem promoters as in
	\cite[\href{https://science.sciencemag.org/content/sci/352/6281/aac7341/F4.large.jpg}{Fig.~3}]{NielsenETAL2016}.
\item
	Cascading
	two \XOR gates, 
	each as in 
	\cite[\href{https://bmcbiol.biomedcentral.com/articles/10.1186/s12915-015-0146-0/figures/5}{Fig.~5}]{Wong2015}
	but
	with two orthogonal heterodimeric transcription activators
	%$\ce{HrpR} \with \ce{HrpS}$ and $\ce{pchA} \with \ce{pchB}$
	(cf.~\S\ref{ss:HrpRS}).
\item
	RNA \AND-gates
	\cite[\href{https://www.nature.com/articles/nature23271/figures/2}{Fig.~2e}]{GreenETAL2017}
	instead of the protein heterodimer.
\item
	\href{https://bit.ly/wsgxfDv}{RNA-only gate}
	inspired by
	\cite[\href{https://www.nature.com/articles/nature23271/figures/2}{Fig.~2e}]{GreenETAL2017}.
	%
	Each input expresses its own mRNA,
	whose
	5' tail opens its RBS stemloop.
	%
	Each input also expresses
	complementary RNAs
	that
	sequester other tails 
	thus closing up other RBS.
	%
	Upon all three inputs,
	all complementary RNAs 
	form a ring,
	so that the 5' tails reopen their RBS.
\item
	Three orthogonal resettable recombinases
	as in 
	\cite[\href{https://www.nature.com/articles/s41598-017-07386-3/figures/2}{Fig.~2g}]{ChiuJiang2017}.
\item
    For completeness, 
    we mention 
    the recombinase-based
    \cite{WeinbergETAL2017}
    % (\url{https://www.addgene.org/87552/}),
    and
    the multi-cell
    \cite{AuslaenderETAL2017}
    one-bit adders.
\end{itemize}


\TODO{Describe the Cas variant}
	



%



%\begin{figure}
%    \centering
%    \includegraphics{circuits/3_input_xor.png}
%    \caption{3 input \XOR logic gate as cascading \XOR gates}
%    \label{f:my_label}
%    \ra{cf. new figure \ref{f:logical-subtractor01}}
%\end{figure}



%\begin{figure}
%    \centering
%    \includegraphics[width = \linewidth]{gene_circuits_xor/P1.png}
%   \includegraphics[width = \linewidth]{gene_circuits_xor/P2.png}
%    \includegraphics[width = \linewidth]{gene_circuits_xor/P3.png}
%    \includegraphics[width = \linewidth]{gene_circuits_xor/P4.png}
%    \caption{Construct of layered \XOR gate. Inducer A, B and C are Ara, Rha and IPTG, which regulate the %activity of pBAD, pRHAB and pTAC respectively. Here the tagged promoters refers to the ones that can be %blocked by the upstream repressor. \hh{This is just a design sketch taht i came up with and drew with %\url{https://sboldesigner.github.io}. For the final report we would use something looked nicer for the %design diagram}
%    \ra{if you keep them online please provide a url and/or upload all relevant files (not just the %figures); also: vector graphics [delete when read]}
%    }
%    \label{f:gene_circuit_xor}
%\end{figure}




%



\section{Simulations}

\TODO{general considerations}


\subsection{PlayerA} \label{s:sim:player}

\url{https://github.com/numpde/ibiocomp/tree/main/code/20201231_SymBio_All/PlayerA}

\TODO{Some details missing}




%

\begin{figure}[hpbt]
	\begin{tabular}{cc|cc}
		\multicolumn{2}{c|}{\ce{\#r_1}} & \multicolumn{2}{c}{\ce{\#r_0}}
		%
		\\
		\hline
		%
		\ce{\#w_A} = 0.1 & \ce{\#w_A} = 10 &
		\ce{\#w_A} = 0.1 & \ce{\#w_A} = 10 
		%
		\\
		%
		\includegraphics[width=0.22\textwidth]{PlayerA/output/response_r0__wA_in=0.1}
		&
		\includegraphics[width=0.22\textwidth]{PlayerA/output/response_r0__wA_in=10}
		&
		\includegraphics[width=0.22\textwidth]{PlayerA/output/response_r1__wA_in=0.1}
		&
		\includegraphics[width=0.22\textwidth]{PlayerA/output/response_r1__wA_in=10}
	\end{tabular}
	%
	\caption{%
		Response surfaces of Player A
		for varying \ce{\#s_1}($\uparrow$) and \ce{\#s_0}($\rightarrow$).
		%
		See \S\ref{s:sim:player}.
		%
		The log-colorbar goes from $10^{-3}$-or-less (white) to $10^1$ (black).
		%
		The yellow contour lines are at $10^t$, $t = 0.8, 0.6, 0.4, \ldots$.
	}
	%
	\label{f:player_response}
\end{figure}





\subsection{Bits 0, 2, 3}

Bits 0, 2, 3 are half-subtractors
which are technically a subfunctionality of Bit~1
(see Table~\ref{t:subtractor1}).
%
We omit closer discussion to focus on Bit~1.


\subsection{Bit 1} \label{s:sim:bit1}

\url{https://github.com/numpde/ibiocomp/tree/main/code/20201231_SymBio_All/Bit1}


\subsubsection*{c2} \label{s:sim:bit1:c2}



\subsubsection*{d1} \label{s:sim:bit1:d1}

Consider Subtractor A.
%
Suppose \ce{s_1 r_1 c_1 = 101},
in which case \ce{d_1 = 0}.
%
Suppose a transcription factor \ce{Y}
promotes expression of \ce{Xis} in the memory subunit
when \ce{d_1 = 1} is intended.
%
%
When \ce{w_B = 1},
access to memory is on.
%
Since we want \ce{d_1 = 0},
the amount of \ce{Y} is low,
but
around the timepoint when 
\ce{w_B} is turned off
and the inputs begin to change,
\ce{Y} can experience a transient bump.
%
This is sufficient to produce \ce{Xis}
and thus to cause an unintended \emph{reset}
towards \ce{d_1 = 1}.
%
We found two ways to mitigate this
in the computational model.
%
One is to condition the expression of \ce{Xis}
on \ce{w_B} also
--
but this requires \ce{w_B} to go down
before other change
and could be sensitive to degradation rates.
%
A more robust way
is a buffer for \ce{Y}
-- a ``shelf'' --
that 
%
i)
has high affinity for \ce{Y};
%
ii)
sequesters 
a large amount of \ce{Y}
(about 30$\times$\href{https://en.wikipedia.org/wiki/EC50}{EC50} of $\ce{Y} \act \ce{Xis}$);
%
iii)
releases \ce{Y} for degradation when \ce{w_A} is on.
%
In this way, 
a transient increase in \ce{Y} is shelved
but a sustained signal 
(while \ce{w_B} is up)
will overflow the shelves and activate 
transcription of \ce{Xis}.
%
%
This is illustrated in Fig.~\ref{f:symbio-d1-tempo}.

\TODO{\ra{Issue: how to implement this ``Shelf''?}}
\ra{%
Note: This seems similar to
\cite[\href{{https://www.pnas.org/content/pnas/104/8/2643/F1.large.jpg?width=800&height=600&carousel=1}}{Fig.~1}]{WeberETAL2007},
also
\cite{NilgiriwalaETAL2014}.
}

%

\begin{figure}[hpbt]
	\begin{tabular}{cc|cc}
        \hline
		\multicolumn{2}{c|}{\ce{\#d_1} with shelving}
		&
		\multicolumn{2}{c}{\ce{\#c_2} with cross-inhibition}
		%
		\\
		\hline
		%
		\ce{\#c_1} = 0.1 & \ce{\#c_1} = 10 &
		\ce{\#c_1} = 0.1 & \ce{\#c_1} = 10 
		%
		\\
		%
		\includegraphics[width=0.22\textwidth]{Bit1/output/response_d1_final__Shelf=30__c1_in=0.1}
		&
		\includegraphics[width=0.22\textwidth]{Bit1/output/response_d1_final__Shelf=30__c1_in=10}
		&
		\includegraphics[width=0.22\textwidth]{Bit1/output/response_c2_final__CI=1__c1_in=0.1}
		&
		\includegraphics[width=0.22\textwidth]{Bit1/output/response_c2_final__CI=1__c1_in=10}
	\end{tabular}
	
	\begin{tabular}{cc|cc}
        \hline
		\multicolumn{2}{c|}{\ce{\#d_1} without shelving}
		&
		\multicolumn{2}{c}{\ce{\#c_2} without cross-inhibition}
		%
		\\
		\hline
		%
		\ce{\#c_1} = 0.1 & \ce{\#c_1} = 10 &
		\ce{\#c_1} = 0.1 & \ce{\#c_1} = 10 
		%
		\\
		%
		\includegraphics[width=0.22\textwidth]{Bit1/output/response_d1_final__Shelf=0__c1_in=0.1}
		&
		\includegraphics[width=0.22\textwidth]{Bit1/output/response_d1_final__Shelf=0__c1_in=10}
		&
		\includegraphics[width=0.22\textwidth]{Bit1/output/response_c2_final__CI=0__c1_in=0.1}
		&
		\includegraphics[width=0.22\textwidth]{Bit1/output/response_c2_final__CI=0__c1_in=10}
	\end{tabular}
	%
	%
	\caption{%
		Response of Subtractor A
		for varying \ce{\#r_1}($\uparrow$) and \ce{\#s_1}($\rightarrow$).
		%
		See \S\ref{s:sim:bit1}.
		%
		The output \ce{\#d_1}
		is measured after a compute-, inter- and readout-phase,
		cf.~Fig.~\ref{f:symbio-d1-tempo}.
		%
		The log-colorbar goes from $10^{-3}$-or-less (white) to $10^1$ (black).
		%
		The yellow contour lines are at $10^t$, $t = 0.8, 0.6, 0.4, \ldots$.
	}
	%
	\label{f:sub_response}
\end{figure}

%

\begin{figure}[phbt]
    \centering
    %
    \begin{overpic}[width=0.99\textwidth]{Bit1/output/response_d1_tempo}
    \put (50, 30) {%
    	\scalebox{0.7}{%
    		\begin{tabular}{c|ccc|c}
    		    & \ce{\#s_1} & \ce{\#r_1} & \ce{\#c_1} & $\approx \ce{\#d_1}$
    		    \\
    		    \hline
    		    \ce{B_1} & 10 & 10 & 10 & 0 
    		    \\
    		    \ce{A_1} & {\color{gray}$0.01$} & {\color{gray}$0.01$} & {\color{gray}$0.01$} & 10 
    		    \\
    		    \ce{B_2} & 10 & 10 & {\color{gray}$0.01$} & 0 
    		    \\
    		    \ce{A_2} & {\color{gray}$0.01$} & {\color{gray}$0.01$} & {\color{gray}$0.01$} & 0
    		    \\
    		\end{tabular}
    	}
    }
    \end{overpic}
    %
    \caption{%
        Subtractor A, bit 1, output \ce{\#d_1}
        across phases \ce{B_1}-\ce{A_1}-\ce{B_2}-\ce{A_2}
        (with interphases).
        %
        The table shows the inputs for each phase.
        %
        As required,
        we have
        $\#\ce{d_1} \approx 10$ 
        during the recall phase \ce{A_1}
        and
        $\#\ce{d_1} \approx 0$ 
        during the recall phase \ce{A_2}.
        %
        Observe the slight delay in the rise of Y
        at the beginning of \ce{B_1} while it is being ``shelved''
        and a sustained supply by ``unshelving'' at the end
        that helps maintaining Xis;
        conversely,
        the leakage of Y is ``shelved''
        during \ce{B_2}
        preventing unwanted expression of Xis.
        %
        See also \S\ref{s:sim:bit1:d1}/p.\pageref{s:sim:bit1:d1}.
    }
    %
    \label{f:symbio-d1-tempo}
\end{figure}

%

\section{Discussion}

\TODO{risks}



\section{Acknowledgments}

\TODO{ack}



%%% BIBLIOGRAPHY %%%
\clearpage

\footnotesize
\bibliographystyle{apalike}
\bibliography{refs}
\normalsize

\clearpage


\section*{Appendix}

\subsection{Subtractor module}

Fig.~\ref{f:logical-subtractor01} 
shows the main principles
behind 
the Subtractor module (bit 0 and bit 1),
but our implementation
differs in the details.
%
In particular, we use a reversible recombinase
for the memory module, see \S\ref{ss:memory}/p.\pageref{ss:memory}.

% https://www.circuit-diagram.org/editor/c/075f8cf9d084400b94f1384cc1c3ba96
\begin{figure}[phbt]
\centering
\includegraphics[width=0.9\linewidth]{circuits/Logical-HalfSubtractor0.svg.pdf}
%
\caption{%
Bit~0 and Bit~1
of
Subtractor~A.
%
The result of the compute unit is written to the memory unit
(e.g.~a bistable switch)
when the signal \ce{w_B} is present.
%
The value of the memory unit is
forwarded to the output
when the signal \ce{w_A} is present.
}
\label{f:logical-subtractor01}
\end{figure}



\subsection{Kinetics}

\TODO{explain guidelines}

\subsubsection*{AmtR}

We assume
the mechanism
\begin{subequations}
\[
	\ce{
		2 \protein{AmtR} + \promoter{AmtR}
		& <=>>
		\protein{AmtR}_2 \with \promoter{AmtR}
	}
	%
	\\
	%
	\ce{
		\promoter{AmtR} 
		& ->
		\promoter{AmtR} + \protein{Output}
	}
\]
\end{subequations}
and
the kinetics:
\begin{subequations}
\[
	\label{e:AmtR_Act}
	%
	\promoter{AmtR} 
	& =
	\frac{1}{1 + k_{\ref{e:AmtR_Act}} \protein{AmtR}^{n_{\ref{e:AmtR_Act}}}}
	\promoter{AmtR}_\text{total}
	%
	% here, k = forward / backward
	%
	,
	\\
	%
	\label{e:AmtR_Out}
	%
	\tfrac{\d}{\d{t}}
	\protein{Output} 
	& =
	k_{\ref{e:AmtR_Out}}
	\promoter{AmtR}
	.
\]
\end{subequations}


\subsubsection*{3OC6}


We write
$\protein{LuxR^\star}$ for $\signal{3OC6\text{-}HSL} \with \protein{LuxR}$
and
$\promoter{Lux^\star}$ for $\protein{LuxR^\star} \with \promoter{Lux}$.
%
%\TODO{
%Let $\protein{LuxR^\star}$ denote 
%the activated form
%$\signal{3OC6}:\protein{LuxR}$.
%}
%
%
We assume the mechanism:
%
\begin{subequations}
\[
	\ce{
		\signal{{3OC6}} + \protein{LuxR}
		& <=>>
		\protein{LuxR^\star}
	}
	\\
	\ce{
		\protein{LuxR^\star} + \promoter{Lux}
		& <=>>
		\promoter{Lux^\star}
	}
	\\
	\ce{
		\promoter{Lux^\star}
		& ->
		\promoter{Lux^\star} + \protein{Output}
	}
	.
\]
\end{subequations}
%
%=======
%	\protein{LuxR^\star} + \protein{RNApol} + \promoter{P_{Lux}}
%	& \longleftrightarrow
%	\protein{LuxR^\star} + \protein{RNApol} : \promoter{P_{Lux}}
%	\\
%	\protein{RNApol} : \promoter{P_{Lux}}
%	& \longrightarrow
%	\protein{RNApol} + \promoter{P_{Lux}} + \protein{Output}
%>>>>>>> c8a0f57bdb0a7b80747263819cbb64562e087765
%\TODO{
%\[
%	\signal{3OC6} + \protein{LuxR}
%	& \longleftrightarrow
%	\protein{LuxR^\star}
%	\\
%	\protein{LuxR^\star} + \protein{RNApol} + \promoter{Lux}
%	& \longleftrightarrow
%	\protein{LuxR^\star} + \protein{RNApol} : \promoter{Lux}
%	\\
%	\protein{RNApol} : \promoter{Lux}
%	& \longrightarrow
%	\protein{RNApol} + \promoter{Lux} + \protein{Output}
%\]
%}
% https://chem.libretexts.org/Bookshelves/Biological_Chemistry/Supplemental_Modules_(Biological_Chemistry)/Enzymes/Enzymatic_Kinetics/Sigmoid_Kinetics
%
%
%
and the kinetics:
%
\begin{subequations}
\[
	\label{e:LuxR_Act}
	%
	\protein{LuxR^\star} 
	& =
	\frac{
		\signal{{3OC6}}
	}{
		k_{\ref{e:LuxR_Act}} + \signal{{3OC6}}
	}
	\protein{LuxR}_\text{total}
%	\TEXT{initially}
%	\protein{LuxR^\star} = 0
	,
	%
	\\
	%%
	\label{e:P_Lux_Act}
	%
	\promoter{Lux^\star} 
	& =
	\frac{
		\protein{LuxR^\star}
	}{
		k_{\ref{e:P_Lux_Act}} + \protein{LuxR^\star}
	}
	\promoter{Lux}_\text{total}
%	\TEXT{initially}
%	\promoter{Lux^\star} = 0
	,
	%
	\\
	%
	\label{e:P_Lux_Out}
	%
	\tfrac{\d}{\d{t}}
	\protein{Output}
	& =
	k_{\ref{e:P_Lux_Out}} \promoter{Lux^\star}
	.
\]
\end{subequations}
%
%
In combination, we have
\[
	\tfrac{\d}{\d{t}}
	\protein{Output}
	=
	k_{\ref{e:P_Lux_Out}}
	\promoter{Lux}_\text{total}
	\frac{
		\signal{} \protein{}_\text{total}
	}{
		k_{\ref{e:LuxR_Act}} k_{\ref{e:P_Lux_Act}}
		+
		k_{\ref{e:P_Lux_Act}} \signal{}
		+
		\signal{} \protein{}_\text{total}
	}
	.
\]


\subsubsection*{Sal}


We write
$
	\promoter{Sal^\star} :=
	\signal{Sal}_4 \with \protein{NahR}_4 \with \promoter{Sal}
$
for the active form.
%
%
We assume the mechanism
%
\begin{subequations}
\[
	\ce{
		\signal{Sal} + 
		\signal{Sal}_{k - 1} \with \protein{NahR}_4 \with \promoter{Sal}
		& <=>>
		\signal{Sal}_k \with \protein{NahR}_4 \with \promoter{Sal}
	}
	,
	\quad \text{k} = 1, 2, 3, 4,
	%
	%
	\\
	%
	%
	\ce{
		\promoter{Sal^\star}
		&
		->
		\promoter{Sal^\star} + \protein{Output}
	}
\]
\end{subequations}
%
%Mechanism (here \protein{NahR^\star} denotes the activated (tetramer) form):
%\[
%	\signal{Sal} + \protein{NahR}
%	& \longleftrightarrow
%	\protein{NahR^\star}
%	 \\
%	 \protein{NahR^\star} + \protein{RNApol} + \promoter{P_{Sal}}
%	& \longleftrightarrow
%	\protein{NahR^\star} + \protein{RNApol} : \promoter{P_{Sal}}
%	\\
%	\protein{RNApol} : \promoter{P_{Sal}}
%	& \longrightarrow
%	\protein{RNApol} + \promoter{P_{Sal}} + \protein{Output}
%	.
%\]
%
%
which we implement as:
%
\begin{subequations}
\[
	\label{e:NahR_Act}
	%
	\promoter{Sal^\star}
	& =
	\frac{
		\signal{}^{n_{\ref{e:NahR_Act}}}
	}{
		k_{\ref{e:NahR_Act}} + \signal{}^{n_{\ref{e:NahR_Act}}}
	}
	\promoter{Sal}_\text{total}
	,
	%
	%
	\\
	%
	%
	\label{e:NahR_Out}
	%
	\tfrac{\d}{\d{t}}
	\protein{Output}
	& =
	k_{\ref{e:NahR_Out}} 
	\promoter{Sal^\star}
	.
\]
\end{subequations}






\clearpage

\section*{Final TODOs}

\TODO{spacing of act and rep}

\TODO{check that Table \ref{t:signals} is correct}


\clearpage

\SHOWTODOS




\leavevmode\vfill{\tiny\color{lightgray}\hfill{\DTMnow}}
\end{document}





